## Investigating 'omics patterns in genes with DNAhm loss
### Samantha Schaffner
### Dec 16, 2022

## Libraries
```{r}
.libPaths(c(.libPaths(), "/mnt/scratch/KoborLab/R_Libs/4.2", "/mnt/scratch/KoborLab/Personal_Folders/sschaffner/Rstudio/R/x86_64-pc-linux-gnu-library/4.0/"))
library(lifecycle, lib.loc="/mnt/scratch/KoborLab/R_Libs/4.2")
library(scales, lib.loc="/mnt/scratch/KoborLab/R_Libs/4.2")
library(vctrs, lib.loc="/mnt/scratch/KoborLab/R_Libs/4.2")

library(reshape2)
library(ggplot2)
library(DescTools)
library(readxl)
library(BiSeq)
library(gridExtra)
library(Gviz)
library(plyr)
library(rtracklayer)
library(GenomicRanges)
library(gplots)
library(RColorBrewer)
library(gprofiler2)
```

## Plotting how these genes behave in all comparisons

## mRNA expression
```{r}
RNA_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_STD_vs_WT_STD.csv")
RNA_WTSE_WTEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_WT_EE_vs_WT_STD.csv")
RNA_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_EE_vs_WT_EE.csv")
RNA_TGSE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_EE_vs_TG_STD.csv")

all.equal(RNA_WTSE_TGSE$mgi_symbol, RNA_WTSE_WTEE$mgi_symbol, RNA_WTEE_TGEE$mgi_symbol, RNA_TGSE_TGEE$mgi_symbol) 
#"'is.NA' value mismatch: 1403 in current 3219 in target"

all.equal(RNA_WTSE_TGSE$ensembl, RNA_WTSE_WTEE$ensembl, RNA_WTEE_TGEE$ensembl, RNA_TGSE_TGEE$ensembl) 
#[1] TRUE

#data frame rows are matched but WTSE TGSE does not have full gene annotation (keep this in mind!)

n <- nrow(RNA_WTSE_TGSE)
RNA_FC <- data.frame(Ensembl=RNA_WTSE_TGSE$ensembl, Gene=RNA_WTSE_TGSE$mgi_symbol,Comparison=c(rep("TGSE_WTSE",n), rep("WTEE_WTSE",n), rep("TGEE_WTEE",n), rep("TGEE_TGSE",n)), log2FC=c(RNA_WTSE_TGSE$log2FoldChange, RNA_WTSE_WTEE$log2FoldChange, RNA_WTEE_TGEE$log2FoldChange, RNA_TGSE_TGEE$log2FoldChange), padj=c(RNA_WTSE_TGSE$padj, RNA_WTSE_WTEE$padj, RNA_WTEE_TGEE$padj, RNA_TGSE_TGEE$padj))

RNA_FC$Comparison <- as.factor(RNA_FC$Comparison)
RNA_FC$Comparison <- reorder.factor(RNA_FC$Comparison, new.order=c("TGSE_WTSE", "WTEE_WTSE", "TGEE_WTEE", "TGEE_TGSE"))

load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/8-multi_omic_integration/overlap/genes_mRNA_DNAhm.RData")
#genes with mRNA and DNAhm change in TGSE vs WTSE comparison
RNA_FC_sub <- RNA_FC[RNA_FC$Gene%in% set17,]
length(unique(RNA_FC_sub$Ensembl)) #6
length(unique(RNA_FC_sub$Gene)) #6 with mgi symbol
for (i in 1:nrow(RNA_FC_sub)){
  if (is.na(RNA_FC_sub$Gene[i])){ RNA_FC_sub$Gene[i] <- RNA_FC_sub$Ensembl[i] }
}

#tile with continuous FC
limit <- max(abs(RNA_FC_sub$log2FC))*c(-1,1)
ggplot(RNA_FC_sub, aes(x=Comparison, y=Gene, fill=log2FC)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=8)) +  scale_fill_distiller(palette="RdBu", direction=1, limit=limit) + labs(title="Genes with DNAhm loss and\ndifferential expression", subtitle="TGSE vs WTSE") + geom_text(label=round(RNA_FC_sub$padj,2))

### plotting counts
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/nCounts_EE_HIP.RData") #use nCounts_mnc50 object. rownames are ensembl genes, colnames sample IDs
sampleInfo <- as.data.frame(sampleInfo)
sampleInfo$group <- paste(sampleInfo$genotype,sampleInfo$environment,sep="")

counts_sub <- nCounts_mnc50[rownames(nCounts_mnc50) %in% RNA_FC_sub$Ensembl,]
ens_to_ext <- RNA_FC_sub[match(rownames(counts_sub), RNA_FC_sub$Ensembl),c("Ensembl","Gene")]
all.equal(rownames(counts_sub), ens_to_ext$Ensembl)
rownames(counts_sub) <- ens_to_ext$Gene
counts_sub <- as.data.frame(t(counts_sub))
all.equal(rownames(counts_sub), sampleInfo$run) #TRUE
counts_sub <- as.data.frame(t(as.matrix(counts_sub)))
counts_scaled <- as.data.frame(apply(counts_sub, 1, function(x) scale(x)))
rownames(counts_scaled) <- paste(sampleInfo$group, rep(1:4,4), sep="")
#counts_scaled$rowmean <- rowMeans(counts_scaled)
counts_scaled$group <- sampleInfo$group

counts_melt <- melt(counts_scaled, id.vars=c("group"))
colnames(counts_melt)[2] <- "Gene"
counts_melt$group <- gsub("STD", "SE", counts_melt$group)
counts_melt$group <- as.factor(counts_melt$group)
counts_melt$group <- reorder.factor(counts_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))
limit <- max(abs(counts_melt$value))*c(-1,1)
```

### DNAhm: beta values
```{r}
#genes with differential DNAhm and expression
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/all_EE_hmC_betas_CpG_10X_auto_var_noout.RData")

lm_WTSE_TGSE_hmC <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/WTSE_TGSE/linear_model/CpG/lm_WTSE_TGSE.csv")
WTSE_TGSE_hmC <- lm_WTSE_TGSE_hmC[lm_WTSE_TGSE_hmC$threshold==TRUE,]
WTSE_TGSE_sub <- WTSE_TGSE_hmC[complete.cases(WTSE_TGSE_hmC$gene) & WTSE_TGSE_hmC$gene %in% set17,]
length(unique(WTSE_TGSE_sub$gene)) #6

nrow(Bvals_hmC_sub <- betas_variable[match(WTSE_TGSE_sub$site, rownames(betas_variable)),]) #6 cytosines
Bvals_hmC_scaled <- as.data.frame(t(apply(Bvals_hmC_sub, 1, function(x) scale(x))))
colnames(Bvals_hmC_scaled) <- colnames(Bvals_hmC_sub)
all.equal(rownames(Bvals_hmC_scaled), WTSE_TGSE_sub$site) #TRUE
#Bvals_hmC_scaled$rowmean <- rowMeans(Bvals_hmC_scaled)
Bvals_hmC_scaled$gene <- WTSE_TGSE_sub$gene
Bvals_hmC_scaled$coord <- rownames(Bvals_hmC_scaled)
Bvals_hmC_scaled$mean_WTSE <- rowMeans(Bvals_hmC_scaled[,1:4])
Bvals_hmC_scaled$mean_WTEE <- rowMeans(Bvals_hmC_scaled[,5:7])
Bvals_hmC_scaled$mean_TGSE <- rowMeans(Bvals_hmC_scaled[,8:10])
Bvals_hmC_scaled$mean_TGEE <- rowMeans(Bvals_hmC_scaled[,11:13])

#adding features
anno <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/cpg_anno_hmC_ChIP.txt")
anno$genomic_anno <- gsub("mm10_genes_introns", "Intron", gsub("mm10_genes_1to5kb", "1to5kb", gsub("enhancers_poised", "Poised Enhancer", gsub("mm10_genes_exonintronboundaries", "Exon-Intron Boundary", gsub("mm10_genes_exons", "Exon", gsub("mm10_genes_intronexonboundaries", "Intron-Exon Boundary", gsub("enhancers_active", "Active Enhancer", gsub("mm10_genes_promoters", "Promoter", gsub("promoters_active", "Active Promoter", gsub("mm10_genes_3UTRs", "3' UTR", gsub("mm10_genes_intergenic", "Intergenic", gsub("mm10_lncrna_gencode", "lncRNA", gsub("mm10_genes_5UTRs", "5' UTR", gsub("mm10_enhancers_fantom", "FANTOM5 Enhancer", anno$genomic_anno))))))))))))))
anno$site <- gsub(": ", ".", anno$site)
anno_sub <- anno[match(Bvals_hmC_scaled$coord, anno$site),]
all.equal(anno_sub$site, Bvals_hmC_scaled$coord) #TRUE
Bvals_hmC_scaled$feature <- anno_sub$genomic_anno

Bvals_hmC_melt <- melt(Bvals_hmC_scaled, id.vars=c("coord","gene","feature", "mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
colnames(Bvals_hmC_melt)[2] <- "Gene"
Bvals_hmC_melt$group <- substr(Bvals_hmC_melt$variable, 1, 4)
Bvals_hmC_melt$group <- as.factor(Bvals_hmC_melt$group)
Bvals_hmC_melt$group <- reorder.factor(Bvals_hmC_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

Bvals_hmC_melt$gene_mean <- Bvals_hmC_melt$mean_WTSE
Bvals_hmC_melt[Bvals_hmC_melt$group=="WTEE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="WTEE","mean_WTEE"]
Bvals_hmC_melt[Bvals_hmC_melt$group=="TGSE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="TGSE","mean_TGSE"]
Bvals_hmC_melt[Bvals_hmC_melt$group=="TGEE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="TGEE","mean_TGEE"]

#tile with continuous hmC level
#use same limit as for mRNA expression, so colour scale can be used for both plots

#add gene name and context to coords
Bvals_hmC_melt$lab <- paste(Bvals_hmC_melt$Gene, Bvals_hmC_melt$feature, sep=" | ")

#subset just to means
nrow(Bvals_hmC_means <- Bvals_hmC_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #78
#library(dplyr)
nrow(Bvals_hmC_means <- distinct(Bvals_hmC_means)) #24
Bvals_hmC_means <- Bvals_hmC_means[order(Bvals_hmC_means$mean_TGSE),]
Bvals_hmC_means$lab <- paste(Bvals_hmC_means$Gene, Bvals_hmC_means$feature, sep=" | ")

labs <- unique(Bvals_hmC_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- Bvals_hmC_means[Bvals_hmC_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    Bvals_hmC_means[Bvals_hmC_means$lab%in%labs[i],] <- anno
  }
}

Bvals_hmC_means$lab <- reorder.factor(Bvals_hmC_means$lab, new.order=unique(as.character(Bvals_hmC_means$lab)))

ggplot(Bvals_hmC_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="DNA\nhydroxymethylation")
```

## Plotting mRNA expression
```{r}
#order same way as DNAhm data
gene_ord <- unique(Bvals_hmC_means$Gene)
counts_melt$Gene <- reorder.factor(counts_melt$Gene, new.order=gene_ord)

ggplot(counts_melt, aes(x=group, y=Gene, fill=value)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12)) +  scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="mRNA\nexpression")
```

# Making a table of differentially hydroxymethylated and expressed genes
```{r}
RNA_sub <- RNA_WTSE_TGSE[RNA_WTSE_TGSE$mgi_symbol %in% set17,]
nrow(RNA_sub) #6
nrow(WTSE_TGSE_sub) #6
write.csv(RNA_sub, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/8-multi_omic_integration/overlap/set17_mRNA_stats.csv", row.names=FALSE)
write.csv(WTSE_TGSE_sub, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/8-multi_omic_integration/overlap/set17_DNAhm_stats.csv", row.names=FALSE)
```

# Overlaps of DNAhm, H3K4me1, and H3K27ac

### H3K27ac: concentration
```{r}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/8-multi_omic_integration/overlap/genes_DNAhm_H3K4me1_H3K27ac.RData")
#"set16" (14 genes)

H3K27ac_conc <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/conc_H3K27ac_2.csv")
H3K27ac_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_TGSE-TGEE_anno.csv")
H3K27ac_TGEE <- H3K27ac_TGEE[H3K27ac_TGEE$coord %in% H3K27ac_conc$coord,]
H3K27ac_conc <- H3K27ac_conc[match(H3K27ac_TGEE$coord, H3K27ac_conc$coord),]
H3K27ac_scaled <- as.data.frame(t(apply(H3K27ac_conc[,3:ncol(H3K27ac_conc)], 1, function(x) scale(x))))
colnames(H3K27ac_scaled) <- colnames(H3K27ac_conc[,3:ncol(H3K27ac_conc)])
H3K27ac_scaled$rowmean <- rowMeans(H3K27ac_scaled)
H3K27ac_scaled$mean_WTSE <- rowMeans(H3K27ac_scaled[,1:4])
H3K27ac_scaled$mean_TGSE <- rowMeans(H3K27ac_scaled[,5:8])
H3K27ac_scaled$mean_WTEE <- rowMeans(H3K27ac_scaled[,9:12])
H3K27ac_scaled$mean_TGEE <- rowMeans(H3K27ac_scaled[,13:16])
H3K27ac_scaled$coord <- H3K27ac_conc$coord
all.equal(H3K27ac_scaled$coord, H3K27ac_TGEE$coord) #TRUE
H3K27ac_scaled$Gene <- H3K27ac_TGEE$SYMBOL

#adding annotations
H3K27ac_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_conspeaks_annotated.csv")
H3K27ac_anno$annotation <- as.factor(H3K27ac_anno$annotation)
H3K27ac_anno$annotation <- reorder.factor(H3K27ac_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))
H3K27ac_anno <- H3K27ac_anno[match(H3K27ac_scaled$coord, H3K27ac_anno$coord),]
all.equal(H3K27ac_anno$coord, H3K27ac_scaled$coord) #TRUE
H3K27ac_scaled$feature <- H3K27ac_anno$annotation
#H3K27ac_scaled <- H3K27ac_scaled[,-1]

#get lists of differentially bound peaks
H3K27ac_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_WTSE-TGSE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K27ac_WTSE_TGSE <- H3K27ac_WTSE_TGSE[-which(H3K27ac_WTSE_TGSE$FDR==min(H3K27ac_WTSE_TGSE$FDR)),]

H3K27ac_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_WTEE-TGEE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K27ac_WTEE_TGEE <- H3K27ac_WTEE_TGEE[-which(H3K27ac_WTEE_TGEE$FDR==min(H3K27ac_WTEE_TGEE$FDR)),]

H3K27ac_scaled <- H3K27ac_scaled[H3K27ac_scaled$coord %in% H3K27ac_WTSE_TGSE[H3K27ac_WTSE_TGSE$FDR<=0.05 & abs(H3K27ac_WTSE_TGSE$Fold)>=0.5,"coord"] & H3K27ac_scaled$Gene %in% set16,]
length(unique(H3K27ac_scaled$Gene)) #14 genes
length(unique(H3K27ac_scaled$coord)) #22 peaks
summary(H3K27ac_scaled$feature)
#  Poised Enhancer   Active Enhancer   Active Promoter          Promoter            5' UTR              Exon            Intron            3' UTR        Downstream 
#                3                 5                 0                 1                 0                 2                11                 0                 0 
#Distal Intergenic 
#                0 

K27ac_melt <- melt(H3K27ac_scaled, id.vars=c("coord","Gene","feature","rowmean","mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
K27ac_melt$group <- substr(K27ac_melt$variable, 1, 4)
K27ac_melt$group <- as.factor(K27ac_melt$group)
K27ac_melt$group <- reorder.factor(K27ac_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

K27ac_melt$gene_mean <- K27ac_melt$mean_WTSE
K27ac_melt[K27ac_melt$group=="WTEE","gene_mean"] <- K27ac_melt[K27ac_melt$group=="WTEE","mean_WTEE"]
K27ac_melt[K27ac_melt$group=="TGSE","gene_mean"] <- K27ac_melt[K27ac_melt$group=="TGSE","mean_TGSE"]
K27ac_melt[K27ac_melt$group=="TGEE","gene_mean"] <- K27ac_melt[K27ac_melt$group=="TGEE","mean_TGEE"]

#tile with continuous H3K27ac level, split by gene feature
#reordering by rowmeans to make nicer visual of high/low DNAm
K27ac_melt <- K27ac_melt[order(K27ac_melt$Gene),]
K27ac_melt$coord <- as.factor(K27ac_melt$coord)
K27ac_melt$coord <- reorder.factor(K27ac_melt$coord, new.order=unique(as.character(K27ac_melt$coord)))

features <- as.character(unique(K27ac_melt$feature))

#add gene name and context
K27ac_melt$lab <- paste(K27ac_melt$Gene, K27ac_melt$feature, sep=" | ")
K27ac_melt$lab <- as.factor(K27ac_melt$lab)
K27ac_melt$lab <- reorder.factor(K27ac_melt$lab, new.order=unique(as.character(K27ac_melt$lab)))

#subset just to means
nrow(K27ac_means <- K27ac_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #352
#library(dplyr)
nrow(K27ac_means <- distinct(K27ac_means)) #88
K27ac_means <- K27ac_means %>% arrange(mean_TGSE)
K27ac_means <- K27ac_means %>% arrange(factor(Gene, levels=unique(as.character(K27ac_means$Gene))))
K27ac_means$lab <- paste(K27ac_means$Gene, K27ac_means$feature, sep=" | ")

labs <- unique(K27ac_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- K27ac_means[K27ac_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    K27ac_means[K27ac_means$lab%in%labs[i],] <- anno
  }
}

K27ac_means$lab <- reorder.factor(K27ac_means$lab, new.order=unique(as.character(K27ac_means$lab)))
```

### H3K4me1: concentration
```{r}
H3K4me1_conc <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/conc_H3K4me1_2.csv")
H3K4me1_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_TGSE-TGEE_anno.csv")
H3K4me1_TGEE <- H3K4me1_TGEE[H3K4me1_TGEE$coord %in% H3K4me1_conc$coord,]
H3K4me1_conc <- H3K4me1_conc[match(H3K4me1_TGEE$coord, H3K4me1_conc$coord),]
H3K4me1_scaled <- as.data.frame(t(apply(H3K4me1_conc[,3:ncol(H3K4me1_conc)], 1, function(x) scale(x))))
colnames(H3K4me1_scaled) <- colnames(H3K4me1_conc[,3:ncol(H3K4me1_conc)])
H3K4me1_scaled$rowmean <- rowMeans(H3K4me1_scaled)
H3K4me1_scaled$mean_WTSE <- rowMeans(H3K4me1_scaled[,1:4])
H3K4me1_scaled$mean_TGSE <- rowMeans(H3K4me1_scaled[,5:8])
H3K4me1_scaled$mean_WTEE <- rowMeans(H3K4me1_scaled[,9:12])
H3K4me1_scaled$mean_TGEE <- rowMeans(H3K4me1_scaled[,13:16])
H3K4me1_scaled$coord <- H3K4me1_conc$coord
all.equal(H3K4me1_scaled$coord, H3K4me1_TGEE$coord) #TRUE
H3K4me1_scaled$Gene <- H3K4me1_TGEE$SYMBOL

#adding annotations
H3K4me1_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_conspeaks_annotated.csv")
H3K4me1_anno$annotation <- as.factor(H3K4me1_anno$annotation)
H3K4me1_anno$annotation <- reorder.factor(H3K4me1_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))
H3K4me1_anno <- H3K4me1_anno[match(H3K4me1_scaled$coord, H3K4me1_anno$coord),]
all.equal(H3K4me1_anno$coord, H3K4me1_scaled$coord) #TRUE
H3K4me1_scaled$feature <- H3K4me1_anno$annotation
#H3K4me1_scaled <- H3K4me1_scaled[,-1]

#get lists of differentially bound peaks
H3K4me1_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_WTSE-TGSE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K4me1_WTSE_TGSE <- H3K4me1_WTSE_TGSE[-which(H3K4me1_WTSE_TGSE$FDR==min(H3K4me1_WTSE_TGSE$FDR)),]

H3K4me1_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_WTEE-TGEE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K4me1_WTEE_TGEE <- H3K4me1_WTEE_TGEE[-which(H3K4me1_WTEE_TGEE$FDR==min(H3K4me1_WTEE_TGEE$FDR)),]

H3K4me1_scaled <- H3K4me1_scaled[H3K4me1_scaled$coord %in% H3K4me1_WTSE_TGSE[H3K4me1_WTSE_TGSE$FDR<=0.05 & abs(H3K4me1_WTSE_TGSE$Fold)>=0.5,"coord"] & H3K4me1_scaled$Gene %in% set16,]
length(unique(H3K4me1_scaled$Gene)) #14 genes
length(unique(H3K4me1_scaled$coord)) #33 peaks
summary(H3K4me1_scaled$feature)
#  Poised Enhancer   Active Enhancer   Active Promoter          Promoter            5' UTR              Exon            Intron            3' UTR        Downstream 
#                1                 8                 1                 2                 0                 0                15                 1                 0 
#Distal Intergenic 
#                5 

K4me1_melt <- melt(H3K4me1_scaled, id.vars=c("coord","Gene","feature","rowmean","mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
K4me1_melt$group <- substr(K4me1_melt$variable, 1, 4)
K4me1_melt$group <- as.factor(K4me1_melt$group)
K4me1_melt$group <- reorder.factor(K4me1_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

K4me1_melt$gene_mean <- K4me1_melt$mean_WTSE
K4me1_melt[K4me1_melt$group=="WTEE","gene_mean"] <- K4me1_melt[K4me1_melt$group=="WTEE","mean_WTEE"]
K4me1_melt[K4me1_melt$group=="TGSE","gene_mean"] <- K4me1_melt[K4me1_melt$group=="TGSE","mean_TGSE"]
K4me1_melt[K4me1_melt$group=="TGEE","gene_mean"] <- K4me1_melt[K4me1_melt$group=="TGEE","mean_TGEE"]

#tile with continuous H3K4me1 level, split by gene feature
#reordering by rowmeans to make nicer visual of high/low DNAm
K4me1_melt <- K4me1_melt[order(K4me1_melt$mean_TGSE),]
K4me1_melt$coord <- as.factor(K4me1_melt$coord)
K4me1_melt$coord <- reorder.factor(K4me1_melt$coord, new.order=unique(as.character(K4me1_melt$coord)))

features <- as.character(unique(K4me1_melt$feature))

#add gene name and context
K4me1_melt$lab <- paste(K4me1_melt$Gene, K4me1_melt$feature, sep=" | ")
K4me1_melt$lab <- as.factor(K4me1_melt$lab)
K4me1_melt$lab <- reorder.factor(K4me1_melt$lab, new.order=unique(as.character(K4me1_melt$lab)))

#subset just to means
nrow(K4me1_means <- K4me1_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #528
#library(dplyr)
nrow(K4me1_means <- distinct(K4me1_means)) #132
K4me1_means <- K4me1_means[order(K4me1_means$mean_TGSE),]
K4me1_means <- K4me1_means %>% arrange(factor(Gene, levels=unique(as.character(K27ac_means$Gene))))
K4me1_means$lab <- paste(K4me1_means$Gene, K4me1_means$feature, sep=" | ")

labs <- unique(K4me1_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- K4me1_means[K4me1_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    K4me1_means[K4me1_means$lab%in%labs[i],] <- anno
  }
}

K4me1_means$lab <- reorder.factor(K4me1_means$lab, new.order=unique(as.character(K4me1_means$lab)))
```

### DNAhm: beta values
```{r}
#genes with differential DNAhm and expression
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/all_EE_hmC_betas_CpG_10X_auto_var_noout.RData")

lm_WTSE_TGSE_hmC <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/WTSE_TGSE/linear_model/CpG/lm_WTSE_TGSE.csv")
WTSE_TGSE_hmC <- lm_WTSE_TGSE_hmC[lm_WTSE_TGSE_hmC$threshold==TRUE,]
WTSE_TGSE_sub <- WTSE_TGSE_hmC[complete.cases(WTSE_TGSE_hmC$gene) & WTSE_TGSE_hmC$gene %in% set16,]
length(unique(WTSE_TGSE_sub$gene)) #14

nrow(Bvals_hmC_sub <- betas_variable[match(WTSE_TGSE_sub$site, rownames(betas_variable)),]) #14 cytosines
Bvals_hmC_scaled <- as.data.frame(t(apply(Bvals_hmC_sub, 1, function(x) scale(x))))
colnames(Bvals_hmC_scaled) <- colnames(Bvals_hmC_sub)
all.equal(rownames(Bvals_hmC_scaled), WTSE_TGSE_sub$site) #TRUE
#Bvals_hmC_scaled$rowmean <- rowMeans(Bvals_hmC_scaled)
Bvals_hmC_scaled$gene <- WTSE_TGSE_sub$gene
Bvals_hmC_scaled$coord <- rownames(Bvals_hmC_scaled)
Bvals_hmC_scaled$mean_WTSE <- rowMeans(Bvals_hmC_scaled[,1:4])
Bvals_hmC_scaled$mean_WTEE <- rowMeans(Bvals_hmC_scaled[,5:7])
Bvals_hmC_scaled$mean_TGSE <- rowMeans(Bvals_hmC_scaled[,8:10])
Bvals_hmC_scaled$mean_TGEE <- rowMeans(Bvals_hmC_scaled[,11:13])

#adding features
anno <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/cpg_anno_hmC_ChIP.txt")
anno$genomic_anno <- gsub("mm10_genes_introns", "Intron", gsub("mm10_genes_1to5kb", "1to5kb", gsub("enhancers_poised", "Poised Enhancer", gsub("mm10_genes_exonintronboundaries", "Exon-Intron Boundary", gsub("mm10_genes_exons", "Exon", gsub("mm10_genes_intronexonboundaries", "Intron-Exon Boundary", gsub("enhancers_active", "Active Enhancer", gsub("mm10_genes_promoters", "Promoter", gsub("promoters_active", "Active Promoter", gsub("mm10_genes_3UTRs", "3' UTR", gsub("mm10_genes_intergenic", "Intergenic", gsub("mm10_lncrna_gencode", "lncRNA", gsub("mm10_genes_5UTRs", "5' UTR", gsub("mm10_enhancers_fantom", "FANTOM5 Enhancer", anno$genomic_anno))))))))))))))
anno$site <- gsub(": ", ".", anno$site)
anno_sub <- anno[match(Bvals_hmC_scaled$coord, anno$site),]
all.equal(anno_sub$site, Bvals_hmC_scaled$coord) #TRUE
Bvals_hmC_scaled$feature <- anno_sub$genomic_anno

Bvals_hmC_melt <- melt(Bvals_hmC_scaled, id.vars=c("coord","gene","feature", "mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
colnames(Bvals_hmC_melt)[2] <- "Gene"
Bvals_hmC_melt$group <- substr(Bvals_hmC_melt$variable, 1, 4)
Bvals_hmC_melt$group <- as.factor(Bvals_hmC_melt$group)
Bvals_hmC_melt$group <- reorder.factor(Bvals_hmC_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

Bvals_hmC_melt$gene_mean <- Bvals_hmC_melt$mean_WTSE
Bvals_hmC_melt[Bvals_hmC_melt$group=="WTEE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="WTEE","mean_WTEE"]
Bvals_hmC_melt[Bvals_hmC_melt$group=="TGSE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="TGSE","mean_TGSE"]
Bvals_hmC_melt[Bvals_hmC_melt$group=="TGEE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="TGEE","mean_TGEE"]

#tile with continuous hmC level
#use same limit as for mRNA expression, so colour scale can be used for both plots

#add gene name and context to coords
Bvals_hmC_melt$lab <- paste(Bvals_hmC_melt$Gene, Bvals_hmC_melt$feature, sep=" | ")

#subset just to means
nrow(Bvals_hmC_means <- Bvals_hmC_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #182
#library(dplyr)
nrow(Bvals_hmC_means <- distinct(Bvals_hmC_means)) #56
Bvals_hmC_means <- Bvals_hmC_means %>% arrange(factor(Gene, levels=unique(as.character(K27ac_means$Gene))))
Bvals_hmC_means$lab <- paste(Bvals_hmC_means$Gene, Bvals_hmC_means$feature, sep=" | ")

labs <- unique(Bvals_hmC_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- Bvals_hmC_means[Bvals_hmC_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    Bvals_hmC_means[Bvals_hmC_means$lab%in%labs[i],] <- anno
  }
}

Bvals_hmC_means$lab <- reorder.factor(Bvals_hmC_means$lab, new.order=unique(as.character(Bvals_hmC_means$lab)))

limit <- max(c(abs(K27ac_melt$gene_mean), abs(K4me1_melt$gene_mean), abs(Bvals_hmC_melt$gene_mean)))*c(-1,1)

ggplot(K4me1_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="H3K4me1")

ggplot(K27ac_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="H3K27ac")

ggplot(Bvals_hmC_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="DNA\nhydroxymethylation")
```

# Overlaps of H3K4me1 and H3K27ac

### H3K27ac: concentration
```{r}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/8-multi_omic_integration/overlap/genes_H3K4me1_H3K27ac.RData")
#"set24" (462 genes)

H3K27ac_conc <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/conc_H3K27ac_2.csv")
H3K27ac_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_TGSE-TGEE_anno.csv")
H3K27ac_TGEE <- H3K27ac_TGEE[H3K27ac_TGEE$coord %in% H3K27ac_conc$coord,]
H3K27ac_conc <- H3K27ac_conc[match(H3K27ac_TGEE$coord, H3K27ac_conc$coord),]
H3K27ac_scaled <- as.data.frame(t(apply(H3K27ac_conc[,3:ncol(H3K27ac_conc)], 1, function(x) scale(x))))
colnames(H3K27ac_scaled) <- colnames(H3K27ac_conc[,3:ncol(H3K27ac_conc)])
H3K27ac_scaled$rowmean <- rowMeans(H3K27ac_scaled)
H3K27ac_scaled$mean_WTSE <- rowMeans(H3K27ac_scaled[,1:4])
H3K27ac_scaled$mean_TGSE <- rowMeans(H3K27ac_scaled[,5:8])
H3K27ac_scaled$mean_WTEE <- rowMeans(H3K27ac_scaled[,9:12])
H3K27ac_scaled$mean_TGEE <- rowMeans(H3K27ac_scaled[,13:16])
H3K27ac_scaled$coord <- H3K27ac_conc$coord
all.equal(H3K27ac_scaled$coord, H3K27ac_TGEE$coord) #TRUE
H3K27ac_scaled$Gene <- H3K27ac_TGEE$SYMBOL

#adding annotations
H3K27ac_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_conspeaks_annotated.csv")
H3K27ac_anno$annotation <- as.factor(H3K27ac_anno$annotation)
H3K27ac_anno$annotation <- reorder.factor(H3K27ac_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))
H3K27ac_anno <- H3K27ac_anno[match(H3K27ac_scaled$coord, H3K27ac_anno$coord),]
all.equal(H3K27ac_anno$coord, H3K27ac_scaled$coord) #TRUE
H3K27ac_scaled$feature <- H3K27ac_anno$annotation
#H3K27ac_scaled <- H3K27ac_scaled[,-1]

#get lists of differentially bound peaks
H3K27ac_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_WTSE-TGSE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K27ac_WTSE_TGSE <- H3K27ac_WTSE_TGSE[-which(H3K27ac_WTSE_TGSE$FDR==min(H3K27ac_WTSE_TGSE$FDR)),]

H3K27ac_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_WTEE-TGEE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K27ac_WTEE_TGEE <- H3K27ac_WTEE_TGEE[-which(H3K27ac_WTEE_TGEE$FDR==min(H3K27ac_WTEE_TGEE$FDR)),]

H3K27ac_scaled <- H3K27ac_scaled[H3K27ac_scaled$coord %in% H3K27ac_WTSE_TGSE[H3K27ac_WTSE_TGSE$FDR<=0.05 & abs(H3K27ac_WTSE_TGSE$Fold)>=0.5,"coord"] & H3K27ac_scaled$Gene %in% set16,]
length(unique(H3K27ac_scaled$Gene)) #14 genes
length(unique(H3K27ac_scaled$coord)) #22 peaks
summary(H3K27ac_scaled$feature)
#  Poised Enhancer   Active Enhancer   Active Promoter          Promoter            5' UTR              Exon            Intron            3' UTR        Downstream 
#                3                 5                 0                 1                 0                 2                11                 0                 0 
#Distal Intergenic 
#                0 

K27ac_melt <- melt(H3K27ac_scaled, id.vars=c("coord","Gene","feature","rowmean","mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
K27ac_melt$group <- substr(K27ac_melt$variable, 1, 4)
K27ac_melt$group <- as.factor(K27ac_melt$group)
K27ac_melt$group <- reorder.factor(K27ac_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

K27ac_melt$gene_mean <- K27ac_melt$mean_WTSE
K27ac_melt[K27ac_melt$group=="WTEE","gene_mean"] <- K27ac_melt[K27ac_melt$group=="WTEE","mean_WTEE"]
K27ac_melt[K27ac_melt$group=="TGSE","gene_mean"] <- K27ac_melt[K27ac_melt$group=="TGSE","mean_TGSE"]
K27ac_melt[K27ac_melt$group=="TGEE","gene_mean"] <- K27ac_melt[K27ac_melt$group=="TGEE","mean_TGEE"]

#tile with continuous H3K27ac level, split by gene feature
#reordering by rowmeans to make nicer visual of high/low DNAm
K27ac_melt <- K27ac_melt[order(K27ac_melt$Gene),]
K27ac_melt$coord <- as.factor(K27ac_melt$coord)
K27ac_melt$coord <- reorder.factor(K27ac_melt$coord, new.order=unique(as.character(K27ac_melt$coord)))

features <- as.character(unique(K27ac_melt$feature))

#add gene name and context
K27ac_melt$lab <- paste(K27ac_melt$Gene, K27ac_melt$feature, sep=" | ")
K27ac_melt$lab <- as.factor(K27ac_melt$lab)
K27ac_melt$lab <- reorder.factor(K27ac_melt$lab, new.order=unique(as.character(K27ac_melt$lab)))

#subset just to means
nrow(K27ac_means <- K27ac_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #352
#library(dplyr)
nrow(K27ac_means <- distinct(K27ac_means)) #88
K27ac_means <- K27ac_means %>% arrange(mean_TGSE)
K27ac_means <- K27ac_means %>% arrange(factor(Gene, levels=unique(as.character(K27ac_means$Gene))))
K27ac_means$lab <- paste(K27ac_means$Gene, K27ac_means$feature, sep=" | ")

labs <- unique(K27ac_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- K27ac_means[K27ac_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    K27ac_means[K27ac_means$lab%in%labs[i],] <- anno
  }
}

K27ac_means$lab <- reorder.factor(K27ac_means$lab, new.order=unique(as.character(K27ac_means$lab)))
```

### H3K4me1: concentration
```{r}
H3K4me1_conc <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/conc_H3K4me1_2.csv")
H3K4me1_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_TGSE-TGEE_anno.csv")
H3K4me1_TGEE <- H3K4me1_TGEE[H3K4me1_TGEE$coord %in% H3K4me1_conc$coord,]
H3K4me1_conc <- H3K4me1_conc[match(H3K4me1_TGEE$coord, H3K4me1_conc$coord),]
H3K4me1_scaled <- as.data.frame(t(apply(H3K4me1_conc[,3:ncol(H3K4me1_conc)], 1, function(x) scale(x))))
colnames(H3K4me1_scaled) <- colnames(H3K4me1_conc[,3:ncol(H3K4me1_conc)])
H3K4me1_scaled$rowmean <- rowMeans(H3K4me1_scaled)
H3K4me1_scaled$mean_WTSE <- rowMeans(H3K4me1_scaled[,1:4])
H3K4me1_scaled$mean_TGSE <- rowMeans(H3K4me1_scaled[,5:8])
H3K4me1_scaled$mean_WTEE <- rowMeans(H3K4me1_scaled[,9:12])
H3K4me1_scaled$mean_TGEE <- rowMeans(H3K4me1_scaled[,13:16])
H3K4me1_scaled$coord <- H3K4me1_conc$coord
all.equal(H3K4me1_scaled$coord, H3K4me1_TGEE$coord) #TRUE
H3K4me1_scaled$Gene <- H3K4me1_TGEE$SYMBOL

#adding annotations
H3K4me1_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_conspeaks_annotated.csv")
H3K4me1_anno$annotation <- as.factor(H3K4me1_anno$annotation)
H3K4me1_anno$annotation <- reorder.factor(H3K4me1_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))
H3K4me1_anno <- H3K4me1_anno[match(H3K4me1_scaled$coord, H3K4me1_anno$coord),]
all.equal(H3K4me1_anno$coord, H3K4me1_scaled$coord) #TRUE
H3K4me1_scaled$feature <- H3K4me1_anno$annotation
#H3K4me1_scaled <- H3K4me1_scaled[,-1]

#get lists of differentially bound peaks
H3K4me1_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_WTSE-TGSE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K4me1_WTSE_TGSE <- H3K4me1_WTSE_TGSE[-which(H3K4me1_WTSE_TGSE$FDR==min(H3K4me1_WTSE_TGSE$FDR)),]

H3K4me1_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_WTEE-TGEE_anno.csv")
#removing Gulp1 (SNCA integration site)
H3K4me1_WTEE_TGEE <- H3K4me1_WTEE_TGEE[-which(H3K4me1_WTEE_TGEE$FDR==min(H3K4me1_WTEE_TGEE$FDR)),]

H3K4me1_scaled <- H3K4me1_scaled[H3K4me1_scaled$coord %in% H3K4me1_WTSE_TGSE[H3K4me1_WTSE_TGSE$FDR<=0.05 & abs(H3K4me1_WTSE_TGSE$Fold)>=0.5,"coord"] & H3K4me1_scaled$Gene %in% set16,]
length(unique(H3K4me1_scaled$Gene)) #14 genes
length(unique(H3K4me1_scaled$coord)) #33 peaks
summary(H3K4me1_scaled$feature)
#  Poised Enhancer   Active Enhancer   Active Promoter          Promoter            5' UTR              Exon            Intron            3' UTR        Downstream 
#                1                 8                 1                 2                 0                 0                15                 1                 0 
#Distal Intergenic 
#                5 

K4me1_melt <- melt(H3K4me1_scaled, id.vars=c("coord","Gene","feature","rowmean","mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
K4me1_melt$group <- substr(K4me1_melt$variable, 1, 4)
K4me1_melt$group <- as.factor(K4me1_melt$group)
K4me1_melt$group <- reorder.factor(K4me1_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

K4me1_melt$gene_mean <- K4me1_melt$mean_WTSE
K4me1_melt[K4me1_melt$group=="WTEE","gene_mean"] <- K4me1_melt[K4me1_melt$group=="WTEE","mean_WTEE"]
K4me1_melt[K4me1_melt$group=="TGSE","gene_mean"] <- K4me1_melt[K4me1_melt$group=="TGSE","mean_TGSE"]
K4me1_melt[K4me1_melt$group=="TGEE","gene_mean"] <- K4me1_melt[K4me1_melt$group=="TGEE","mean_TGEE"]

#tile with continuous H3K4me1 level, split by gene feature
#reordering by rowmeans to make nicer visual of high/low DNAm
K4me1_melt <- K4me1_melt[order(K4me1_melt$mean_TGSE),]
K4me1_melt$coord <- as.factor(K4me1_melt$coord)
K4me1_melt$coord <- reorder.factor(K4me1_melt$coord, new.order=unique(as.character(K4me1_melt$coord)))

features <- as.character(unique(K4me1_melt$feature))

#add gene name and context
K4me1_melt$lab <- paste(K4me1_melt$Gene, K4me1_melt$feature, sep=" | ")
K4me1_melt$lab <- as.factor(K4me1_melt$lab)
K4me1_melt$lab <- reorder.factor(K4me1_melt$lab, new.order=unique(as.character(K4me1_melt$lab)))

#subset just to means
nrow(K4me1_means <- K4me1_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #528
#library(dplyr)
nrow(K4me1_means <- distinct(K4me1_means)) #132
K4me1_means <- K4me1_means[order(K4me1_means$mean_TGSE),]
K4me1_means <- K4me1_means %>% arrange(factor(Gene, levels=unique(as.character(K27ac_means$Gene))))
K4me1_means$lab <- paste(K4me1_means$Gene, K4me1_means$feature, sep=" | ")

labs <- unique(K4me1_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- K4me1_means[K4me1_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    K4me1_means[K4me1_means$lab%in%labs[i],] <- anno
  }
}

K4me1_means$lab <- reorder.factor(K4me1_means$lab, new.order=unique(as.character(K4me1_means$lab)))
```

### DNAhm: beta values
```{r}
#genes with differential DNAhm and expression
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/all_EE_hmC_betas_CpG_10X_auto_var_noout.RData")

lm_WTSE_TGSE_hmC <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/WTSE_TGSE/linear_model/CpG/lm_WTSE_TGSE.csv")
WTSE_TGSE_hmC <- lm_WTSE_TGSE_hmC[lm_WTSE_TGSE_hmC$threshold==TRUE,]
WTSE_TGSE_sub <- WTSE_TGSE_hmC[complete.cases(WTSE_TGSE_hmC$gene) & WTSE_TGSE_hmC$gene %in% set16,]
length(unique(WTSE_TGSE_sub$gene)) #14

nrow(Bvals_hmC_sub <- betas_variable[match(WTSE_TGSE_sub$site, rownames(betas_variable)),]) #14 cytosines
Bvals_hmC_scaled <- as.data.frame(t(apply(Bvals_hmC_sub, 1, function(x) scale(x))))
colnames(Bvals_hmC_scaled) <- colnames(Bvals_hmC_sub)
all.equal(rownames(Bvals_hmC_scaled), WTSE_TGSE_sub$site) #TRUE
#Bvals_hmC_scaled$rowmean <- rowMeans(Bvals_hmC_scaled)
Bvals_hmC_scaled$gene <- WTSE_TGSE_sub$gene
Bvals_hmC_scaled$coord <- rownames(Bvals_hmC_scaled)
Bvals_hmC_scaled$mean_WTSE <- rowMeans(Bvals_hmC_scaled[,1:4])
Bvals_hmC_scaled$mean_WTEE <- rowMeans(Bvals_hmC_scaled[,5:7])
Bvals_hmC_scaled$mean_TGSE <- rowMeans(Bvals_hmC_scaled[,8:10])
Bvals_hmC_scaled$mean_TGEE <- rowMeans(Bvals_hmC_scaled[,11:13])

#adding features
anno <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/cpg_anno_hmC_ChIP.txt")
anno$genomic_anno <- gsub("mm10_genes_introns", "Intron", gsub("mm10_genes_1to5kb", "1to5kb", gsub("enhancers_poised", "Poised Enhancer", gsub("mm10_genes_exonintronboundaries", "Exon-Intron Boundary", gsub("mm10_genes_exons", "Exon", gsub("mm10_genes_intronexonboundaries", "Intron-Exon Boundary", gsub("enhancers_active", "Active Enhancer", gsub("mm10_genes_promoters", "Promoter", gsub("promoters_active", "Active Promoter", gsub("mm10_genes_3UTRs", "3' UTR", gsub("mm10_genes_intergenic", "Intergenic", gsub("mm10_lncrna_gencode", "lncRNA", gsub("mm10_genes_5UTRs", "5' UTR", gsub("mm10_enhancers_fantom", "FANTOM5 Enhancer", anno$genomic_anno))))))))))))))
anno$site <- gsub(": ", ".", anno$site)
anno_sub <- anno[match(Bvals_hmC_scaled$coord, anno$site),]
all.equal(anno_sub$site, Bvals_hmC_scaled$coord) #TRUE
Bvals_hmC_scaled$feature <- anno_sub$genomic_anno

Bvals_hmC_melt <- melt(Bvals_hmC_scaled, id.vars=c("coord","gene","feature", "mean_WTSE","mean_TGSE","mean_WTEE","mean_TGEE"))
colnames(Bvals_hmC_melt)[2] <- "Gene"
Bvals_hmC_melt$group <- substr(Bvals_hmC_melt$variable, 1, 4)
Bvals_hmC_melt$group <- as.factor(Bvals_hmC_melt$group)
Bvals_hmC_melt$group <- reorder.factor(Bvals_hmC_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

Bvals_hmC_melt$gene_mean <- Bvals_hmC_melt$mean_WTSE
Bvals_hmC_melt[Bvals_hmC_melt$group=="WTEE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="WTEE","mean_WTEE"]
Bvals_hmC_melt[Bvals_hmC_melt$group=="TGSE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="TGSE","mean_TGSE"]
Bvals_hmC_melt[Bvals_hmC_melt$group=="TGEE","gene_mean"] <- Bvals_hmC_melt[Bvals_hmC_melt$group=="TGEE","mean_TGEE"]

#tile with continuous hmC level
#use same limit as for mRNA expression, so colour scale can be used for both plots

#add gene name and context to coords
Bvals_hmC_melt$lab <- paste(Bvals_hmC_melt$Gene, Bvals_hmC_melt$feature, sep=" | ")

#subset just to means
nrow(Bvals_hmC_means <- Bvals_hmC_melt[,c("coord","Gene","feature","group","gene_mean","mean_TGSE","lab")]) #182
#library(dplyr)
nrow(Bvals_hmC_means <- distinct(Bvals_hmC_means)) #56
Bvals_hmC_means <- Bvals_hmC_means %>% arrange(factor(Gene, levels=unique(as.character(K27ac_means$Gene))))
Bvals_hmC_means$lab <- paste(Bvals_hmC_means$Gene, Bvals_hmC_means$feature, sep=" | ")

labs <- unique(Bvals_hmC_means$lab)
for (i in 1:length(labs)){
  #print(paste(i, labs[i]))
  anno <- Bvals_hmC_means[Bvals_hmC_means$lab%in%labs[i],]
  if (length(unique(anno$coord))>1){
    for (j in 1:length(unique(anno$coord))){
      anno$lab[(j*4-3):(j*4)] <- paste(anno$lab[(j*4-3):(j*4)], j, sep=" | ")
    }
    Bvals_hmC_means[Bvals_hmC_means$lab%in%labs[i],] <- anno
  }
}

Bvals_hmC_means$lab <- reorder.factor(Bvals_hmC_means$lab, new.order=unique(as.character(Bvals_hmC_means$lab)))

limit <- max(c(abs(K27ac_melt$gene_mean), abs(K4me1_melt$gene_mean), abs(Bvals_hmC_melt$gene_mean)))*c(-1,1)

ggplot(K4me1_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="H3K4me1")

ggplot(K27ac_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="H3K27ac")

ggplot(Bvals_hmC_means, aes(x=group, y=lab, fill=gene_mean)) + geom_tile() + theme_bw() + theme(axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.x=element_text(angle=90, hjust=1), plot.title=element_text(hjust=0.5, vjust=1, size=12), plot.subtitle = element_text(hjust=0.5, size=10)) + scale_fill_distiller(palette="RdBu", direction=-1, limit=limit, name="Row\nZ-score") + labs(title="DNA\nhydroxymethylation")
```

---