## Exploring genes with DNAhm loss in WTSE vs TGSE comparison
### Samantha Schaffner
### Dec 19, 2022

## Libraries
```{r}
.libPaths(c(.libPaths(), "/mnt/scratch/KoborLab/R_Libs/4.2", "/mnt/scratch/KoborLab/Personal_Folders/sschaffner/Rstudio/R/x86_64-pc-linux-gnu-library/4.0/"))
library(lifecycle, lib.loc="/mnt/scratch/KoborLab/R_Libs/4.2")
library(scales, lib.loc="/mnt/scratch/KoborLab/R_Libs/4.2")
library(vctrs, lib.loc="/mnt/scratch/KoborLab/R_Libs/4.2")

library(reshape2)
library(ggplot2)
library(DescTools)
library(readxl)
#library(BiSeq)
library(gridExtra)
library(Gviz)
library(plyr)
library(rtracklayer)
library(GenomicRanges)
library(gplots)
library(RColorBrewer)
library(gprofiler2)
```

## Reading in candidate gene information
```{r}
#genes with differential DNAhm, H3K4me1, and H3K27ac
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/8-multi_omic_integration/overlap/genes_DNAhm_H3K4me1_H3K27ac_SE.RData")
#"set16" (14 genes)

set16
#"Mgat5"    "Cacna1c"  "Col4a1"   "Dennd2a"  "Notch2"   "Adamts20" "Kcnq2"    "Atp2b2"   "Sorbs1"   "Kazn"     "Arid5b"   "Vps35l"   "Rnf165"   "Ip6k1"   
```

## Plotting how these genes behave in all comparisons

### Expression: read counts
The object "nCounts_mnc50" contains info for genes that had at least 50 reads on average, which is what Thomas used for differential expression.

To convert to reads per kilobase million (RPKM), the following steps can be applied (taken from https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/):

1. Count up the total reads in a sample and divide that number by 1,000,000 – this is our “per million” scaling factor.

2. Divide the read counts by the “per million” scaling factor. This normalizes for sequencing depth, giving you reads per million (RPM)

3. Divide the RPM values by the length of the gene, in kilobases. This gives you RPKM.

```{r}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/nCounts_EE_HIP.RData") #use nCounts_mnc50 object. rownames are ensembl genes, colnames sample IDs
sampleInfo <- as.data.frame(sampleInfo)
sampleInfo$group <- paste(sampleInfo$genotype,sampleInfo$environment,sep="")

#get ensembl gene IDs to cross reference counts matrix
mRNA_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_EE_vs_TG_STD.csv")
mRNA_TGEE <- mRNA_TGEE[mRNA_TGEE$mgi_symbol %in% set16,]
mRNA_TGEE <- mRNA_TGEE[complete.cases(mRNA_TGEE$mgi_symbol),]
ensgenes <- mRNA_TGEE$ensembl
all(ensgenes %in% rownames(nCounts_mnc50)) #TRUE

#convert to RPKM
#scaling_factor <- colSums(nCounts_mnc50)/1000000
#nRPM <- lapply(1:ncol(nCounts_mnc50), function(x) { nCounts_mnc50[,x]/scaling_factor[x] } )
#nRPM <- do.call(cbind, nRPM)
#get gene lengths from biomaRt
#mart <- useMart("ENSEMBL_MART_ENSEMBL", dataset="mmusculus_gene_ensembl")
#gene_lengths <- getBM(mart=mart, filters="ensembl_gene_id", attributes=c("ensembl_gene_id","start_position","end_position"), values=rownames(nRPM))
#gene_lengths$length_kb <- (gene_lengths$end_position - gene_lengths$start_position)/1000
#nRPM <- nRPM[match(gene_lengths$ensembl_gene_id, rownames(nRPM)),]
#all.equal(rownames(nRPM), gene_lengths$ensembl_gene_id)#TRUE
#nRPKM <- nRPM/gene_lengths$length_kb
#colnames(nRPKM) <- colnames(nCounts_mnc50)
#save(nRPKM, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/nRPKM_EE_HIP_SS.RData")
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/nRPKM_EE_HIP_SS.RData")


counts_sub <- nRPKM[rownames(nRPKM) %in% ensgenes,] #12 genes
counts_sub <- as.data.frame(t(counts_sub))
all.equal(rownames(counts_sub), sampleInfo$run) #TRUE
mRNA_TGEE <- mRNA_TGEE[match(colnames(counts_sub), mRNA_TGEE$ensembl),]
all.equal(mRNA_TGEE$ensembl, colnames(counts_sub)) #TRUE
colnames(counts_sub) <- mRNA_TGEE$mgi_symbol
counts_sub$group <- sampleInfo$group

mRNA_melt <- melt(counts_sub, id.vars="group")
colnames(mRNA_melt)[2] <- "Gene"
mRNA_melt$group <- gsub("STD", "SE", mRNA_melt$group)
mRNA_melt$group <- as.factor(mRNA_melt$group)
mRNA_melt$group <- reorder.factor(mRNA_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

#plot
Mgat5 <- mRNA_melt[mRNA_melt$Gene=="Mgat5",]
ggplot(Mgat5, aes(x=group,y=value,col=group)) + scale_colour_manual(values=c("black","plum3","gray60","palegreen3")) + theme_classic() + facet_wrap(~Gene) + geom_point() + geom_boxplot() + ylab("nRPKM") + theme(axis.title.x=element_blank())

for (i in 1:length(set16)){
  #png(paste("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/figures/Fig3/Fig3D/mRNA_", set16[i], ".png", sep=""), width=360, height=227)
  ggplot(mRNA_melt[mRNA_melt$Gene==set16[i],], aes(x=group,y=value,col=group)) + scale_colour_manual(values=c("black","plum3","gray60","palegreen3")) + theme_classic() + facet_wrap(~Gene) + geom_point() + geom_boxplot() + ylab("nRPKM") + theme(axis.title.x=element_blank())
  #dev.off()
}


#check stats
RNA_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_STD_vs_WT_STD.csv")
RNA_WTSE_TGSE$threshold <- (RNA_WTSE_TGSE$padj<=0.15 & abs(RNA_WTSE_TGSE$log2FoldChange)>=0.3)
RNA_WTSE_WTEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_WT_EE_vs_WT_STD.csv")
RNA_WTSE_WTEE$threshold <- (RNA_WTSE_WTEE$padj<=0.15 & abs(RNA_WTSE_WTEE$log2FoldChange)>=0.3)
RNA_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_EE_vs_WT_EE.csv")
RNA_WTEE_TGEE$threshold <- (RNA_WTEE_TGEE$padj<=0.15 & abs(RNA_WTEE_TGEE$log2FoldChange)>=0.3)
RNA_TGSE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_TG_EE_vs_TG_STD.csv")
RNA_TGSE_TGEE$threshold <- (RNA_TGSE_TGEE$padj<=0.15 & abs(RNA_TGSE_TGEE$log2FoldChange)>=0.3)
RNA_interaction <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/6-processed_RNAseq_data/mmus_hippo12m_interactionTerm.csv")
RNA_interaction$threshold <- (RNA_interaction$padj<=0.15 & abs(RNA_interaction$log2FoldChange)>=0.3)

RNA_WTSE_TGSE[RNA_WTSE_TGSE$mgi_symbol%in%set16 & RNA_WTSE_TGSE$threshold==TRUE,] #none
RNA_WTSE_WTEE[RNA_WTSE_WTEE$mgi_symbol%in%set16 & RNA_WTSE_WTEE$threshold==TRUE,] #none
RNA_WTSE_TGSE[RNA_WTEE_TGEE$mgi_symbol%in%set16 & RNA_WTEE_TGEE$threshold==TRUE,] #none
RNA_WTSE_TGSE[RNA_TGSE_TGEE$mgi_symbol%in%set16 & RNA_TGSE_TGEE$threshold==TRUE,] #none
RNA_interaction[RNA_interaction$mgi_symbol%in%set16 & RNA_interaction$threshold==TRUE,] #none
```

### DNAhm: beta values
```{r}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/all_EE_hmC_betas_CpG_10X_auto_var_noout.RData")

anno <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/hydroxy/CpG/cpg_anno_hmC_ChIP.txt")
anno$site <- gsub(": ", ".", anno$site)
all(anno$site==rownames(betas_variable)) #TRUE
betas_variable <- cbind(betas_variable, anno[,2:5])

Bvals_hmC_sub <- betas_variable[betas_variable$gene %in% set16,] #34 cytosines
Bvals_hmC_sub$coord <- rownames(Bvals_hmC_sub)
colnames(Bvals_hmC_sub)[16] <- "feature"
Bvals_hmC_sub$feature <- gsub("mm10_genes_","",Bvals_hmC_sub$feature)

Bvals_hmC_melt <- melt(Bvals_hmC_sub, id.vars=c("coord","gene","tx","feature","cgi_anno"))
colnames(Bvals_hmC_melt)[2] <- "Gene"
Bvals_hmC_melt$group <- substr(Bvals_hmC_melt$variable, 1, 4)
Bvals_hmC_melt$group <- as.factor(Bvals_hmC_melt$group)
Bvals_hmC_melt$group <- reorder.factor(Bvals_hmC_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))
Bvals_hmC_melt$feature <- as.factor(Bvals_hmC_melt$feature)
unique(Bvals_hmC_melt$feature)
Bvals_hmC_melt$feature <- reorder.factor(Bvals_hmC_melt$feature, new.order=c("enhancers_poised", "enhancers_active","exonintronboundaries","intronexonboundaries", "3UTRs"))
```


### Plotting Mgat5
```{r}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/8-multi_omic_integration/overlap/mm10_bed.RData")
Mgat5 <- mm10_bed[mm10_bed$V13=="Mgat5",]
chr <- unique(Mgat5$V3)
min <- min(Mgat5$V5) - 5000
max <- max(Mgat5$V5) + 5000

################## DNAhm
fdat_hmC <- data.frame(site=rownames(betas_variable), chr=sapply(1:nrow(betas_variable), function(x) unlist(strsplit(rownames(betas_variable)[x], split="\\."))[[1]]), 
                       pos=sapply(1:nrow(betas_variable), function(x) unlist(strsplit(rownames(betas_variable)[x], split="\\."))[[2]]))
fdat_hmC$pos <- as.integer(fdat_hmC$pos)
#UCSC annotation in mm10
#chr1:127205018-127488336

subbed_fdat_hmC <- fdat_hmC[fdat_hmC$chr==chr & fdat_hmC$pos>=(127205018-5000) & fdat_hmC$pos<=(127488336+5000),] #7 sites

######## check stats for differential hydroxymethylation

### TGSE vs WTSE
lm_WTSE_TGSE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/WTSE_TGSE/linear_model/CpG/lm_WTSE_TGSE.csv")
lm_WTSE_TGSE[lm_WTSE_TGSE$site %in% subbed_fdat_hmC$site & lm_WTSE_TGSE$threshold==TRUE,]
#      Estimate Std..Error   t.value     Pr...t..         z           site         DB       adjP threshold DNAm_change  gene                 tx
#411 -0.3304225 0.05862279 -5.636417 0.0003190762 -3.868263 chr1.127378802 -0.3304225 0.02339050      TRUE    Decrease  <NA>               <NA>
#413 -0.2275763 0.06355910 -3.580546 0.0059256961 -2.896087 chr1.127397681 -0.2275763 0.09811509      TRUE    Decrease Mgat5 ENSMUST00000171405
#                       genomic_anno       cgi_anno
#411              mm10_genes_introns mm10_cpg_inter
#413 mm10_genes_exonintronboundaries mm10_cpg_inter

### WTEE vs WTSE
lm_WTSE_WTEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/WTSE_WTEE/linear_model/CpG/lm_WTSE_WTEE.csv")
lm_WTSE_WTEE$threshold <- (lm_WTSE_WTEE$adjP<=0.15 & abs(lm_WTSE_WTEE$DB)>=0.1)
lm_WTSE_WTEE[lm_WTSE_WTEE$site %in% subbed_fdat_hmC$site & lm_WTSE_WTEE$threshold==TRUE,]
#none

### TGEE vs TGSE
lm_TGSE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/TGSE_TGEE/linear_model/CpG/lm_TGSE_TGEE.csv")
lm_TGSE_TGEE$threshold <- (lm_TGSE_TGEE$adjP<=0.15 & abs(lm_TGSE_TGEE$DB)>=0.1)
lm_TGSE_TGEE[lm_TGSE_TGEE$site %in% subbed_fdat_hmC$site & lm_TGSE_TGEE$threshold==TRUE,]
#none

### TGEE vs WTEE
lm_WTEE_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/hydroxy/WTEE_TGEE/linear_model/CpG/lm_WTEE_TGEE.csv")
lm_WTEE_TGEE[lm_WTEE_TGEE$site %in% subbed_fdat_hmC$site & lm_WTEE_TGEE$threshold==TRUE,]
#     Estimate Std..Error  t.value    Pr...t..        z           site        DB      adjP threshold DNAm_change  gene                 tx
#412 0.2961855 0.08042634 3.682692 0.005054322 2.954342 chr1.127397640 0.2961855 0.1021417      TRUE    Increase Mgat5 ENSMUST00000171405
#                       genomic_anno       cgi_anno
#412 mm10_genes_exonintronboundaries mm10_cpg_inter

######## 
anno[anno$site %in% subbed_fdat_hmC$site,c("site","genomic_anno","cgi_anno")]
#              site                    genomic_anno        cgi_anno
#407 chr1.127203880               mm10_genes_1to5kb mm10_cpg_shores
#408 chr1.127203945               mm10_genes_1to5kb mm10_cpg_shores
#409 chr1.127336407              mm10_genes_introns  mm10_cpg_inter
#410 chr1.127378767              mm10_genes_introns  mm10_cpg_inter
#411 chr1.127378802              mm10_genes_introns  mm10_cpg_inter
#412 chr1.127397640 mm10_genes_exonintronboundaries  mm10_cpg_inter
#413 chr1.127397681 mm10_genes_exonintronboundaries  mm10_cpg_inter

betas.hmC <- as.data.frame(t(as.matrix(betas_variable[rownames(betas_variable) %in% subbed_fdat_hmC$site,1:13])))
betas.hmC$Group <- substr(rownames(betas.hmC), start=1, stop=4)

betas.hmC.WTSE <- betas.hmC[betas.hmC$Group == "WTSE",]
betas.hmC.WTSE <- betas.hmC.WTSE[complete.cases(betas.hmC.WTSE),]
betas.hmC.WTSE$Group <- NULL
betas.hmC.WTSE <- as.data.frame(t(betas.hmC.WTSE))
betas.hmC.WTSE$symbol <- rownames(betas.hmC.WTSE)

betas.hmC.WTEE <- betas.hmC[betas.hmC$Group == "WTEE",]
betas.hmC.WTEE <- betas.hmC.WTEE[complete.cases(betas.hmC.WTEE),]
betas.hmC.WTEE$Group <- NULL
betas.hmC.WTEE <- as.data.frame(t(betas.hmC.WTEE))
betas.hmC.WTEE$symbol <- rownames(betas.hmC.WTEE)

betas.hmC.TGSE <- betas.hmC[betas.hmC$Group == "TGSE",]
betas.hmC.TGSE <- betas.hmC.TGSE[complete.cases(betas.hmC.TGSE),]
betas.hmC.TGSE$Group <- NULL
betas.hmC.TGSE <- as.data.frame(t(betas.hmC.TGSE))
betas.hmC.TGSE$symbol <- rownames(betas.hmC.TGSE)

betas.hmC.TGEE <- betas.hmC[betas.hmC$Group == "TGEE",]
betas.hmC.TGEE <- betas.hmC.TGEE[complete.cases(betas.hmC.TGEE),]
betas.hmC.TGEE$Group <- NULL
betas.hmC.TGEE <- as.data.frame(t(betas.hmC.TGEE))
betas.hmC.TGEE$symbol <- rownames(betas.hmC.TGEE)

Group_betas_hmC <- join(betas.hmC.WTSE, betas.hmC.WTEE)
Group_betas_hmC <- join(Group_betas_hmC, betas.hmC.TGSE)
Group_betas_hmC <- join(Group_betas_hmC, betas.hmC.TGEE)
rownames(Group_betas_hmC) <- Group_betas_hmC$symbol
Mgat5_fdat <- subbed_fdat_hmC
colnames(Mgat5_fdat)[1] <- "symbol"
Mgat5_fdat$genome <- "mm10"
Group_betas_hmC <- join(Group_betas_hmC, Mgat5_fdat)
rownames(Group_betas_hmC) <- Group_betas_hmC$symbol
colnames(Group_betas_hmC)
Group_betas_hmC <- Group_betas_hmC[, c(5, 15:17, 1:4, 6:14)]
colnames(Group_betas_hmC)
```

Producing tracks and making figure
```{r}
# Chromosome Track
Mgat5_itrack <- IdeogramTrack(genome = unique(Group_betas_hmC$genome), chromosome = unique(as.character(Group_betas_hmC$chr)))
# Basepair map track
Mgat5_gtrack <- GenomeAxisTrack(add53 = TRUE, add35 = TRUE, labelPos = "alternating")

#intron/exon track
knownGenes <- UcscTrack(genome = "mm10", chromosome = unique(as.character(Group_betas_hmC$chr)), 
                        track = "knownGene", from = min(Group_betas_hmC$pos), to = max(Group_betas_hmC$pos),
                        trackType = "GeneRegionTrack", 
                        rstarts = "exonStarts", rends = "exonEnds", 
                        gene = "name", symbol = "name", 
                        transcript = "name", strand = "strand", 
                        fill = "grey", name = "UCSC Genes")

#hmC track
Mgat5_dTrack_hmC <- DataTrack(data = Group_betas_hmC[,c(9:11,15:17,5:8,12:14)], name = "DNAhm beta Value", start = Group_betas_hmC$pos, width = 1, chromosome = unique(as.character(Group_betas_hmC$chr)), genome = unique(Group_betas_hmC$genome), groups = rep(c("WTEE", "TGEE", "WTSE","TGSE"), c(3,3,4,3)), type = c("boxplot"), col=c("grey60", "palegreen3", "black","plum3"), cex.legend = 1, ylim=c(0,1), col.boxplotFrame="transparent",box.width = 4, box.ratio=0.8)

#plotting everything
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC), from = 127205018-5000, to = 127488336+5000, sizes = c(0.05,0.1,0.05,0.2), col.axis="black", background.title="white")

#chr1:127205018-127488336

```

# Plotting H3K4me1 at Mgat5
For a representative example.

```{r Mgat5 plot, eval=F}
################## H3K4me1

######## check stats for differential H3K4me1
WTSE_TGSE_H3K4me1 <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_sig_only/H3K4me1/H3K4me1_WTSE-TGSE_cons-2_FDR0_05_FC0_5_annot.tsv")
WTSE_TGSE_H3K4me1[WTSE_TGSE_H3K4me1$SYMBOL=="Mgat5",c("chr","start","end","Fold", "FDR")]
#      chr     start       end  Fold    FDR
#3337 chr1 127355656 127355959 -1.87 0.0331
#4268 chr1 127329211 127331204 -0.70 0.0449

#none for WTSE-WTEE, TGSE-TGEE

WTEE_TGEE_H3K4me1 <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_sig_only/H3K4me1/H3K4me1_WTEE-TGEE_cons-2_FDR0_05_FC0_5_annot.tsv")
WTEE_TGEE_H3K4me1[WTEE_TGEE_H3K4me1$SYMBOL=="Mgat5" & complete.cases(WTEE_TGEE_H3K4me1$SYMBOL),c("chr","start","end","Fold", "FDR")]
#      chr     start       end Fold    FDR
#3170 chr1 127222031 127225276 0.59 0.0267
#4915 chr1 127400178 127400661 1.49 0.0486

#adding annotations
H3K4me1_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_conspeaks_annotated.csv")
H3K4me1_anno$annotation <- as.factor(H3K4me1_anno$annotation)
H3K4me1_anno$annotation <- reorder.factor(H3K4me1_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))

######## 

#UCSC annotation in mm10
#chr1:127205018-127488336

nrow(subbed_fdat_K4me1 <- H3K4me1_anno[H3K4me1_anno$seqnames=="chr1" & H3K4me1_anno$end>=127205018-5000 & H3K4me1_anno$start<=127488336+5000,]) #64 peaks
sort(summary(subbed_fdat_K4me1$annotation))
#           5' UTR        Downstream Distal Intergenic   Active Promoter          Promoter            3' UTR   Active Enhancer              Exon 
#                0                 0                 0                 1                 1                 2                 5                 6 
#  Poised Enhancer            Intron 
#               19                30

# Chromosome Track
Mgat5_itrack <- IdeogramTrack(genome = "mm10", chromosome = "chr1")
# Basepair map track
Mgat5_gtrack <- GenomeAxisTrack(add53 = TRUE, add35 = TRUE, labelPos = "alternating")

#intron/exon track
#UCSC coords chr11:69098948-69109960
knownGenes <- UcscTrack(genome = "mm10", chromosome = "chr1", 
                        track = "knownGene", from = 127205018-5000, to = 127488336+5000,
                        trackType = "GeneRegionTrack", 
                        rstarts = "exonStarts", rends = "exonEnds", 
                        gene = "name", symbol = "name", 
                        transcript = "name", strand = "strand", 
                        fill = "grey", name = "UCSC Genes")

#### H3K4me1 tracks
bw_H3K4me1_WTSE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K4me1/merged_c_WTSE.bw", as="GRanges")
bw_H3K4me1_WTEE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K4me1/merged_c_WTEE.bw", as="GRanges")
bw_H3K4me1_TGSE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K4me1/merged_c_TGSE.bw", as="GRanges")
bw_H3K4me1_TGEE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K4me1/merged_c_TGEE.bw", as="GRanges")

#making trakcs
#optional argument: ylim=c(min,max)
H3K4me1_WTSE_track <- DataTrack(range=bw_H3K4me1_WTSE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="black",col.histogram="black",
                         name="H3K4me1 WTSE",
                         type="histogram", alpha=0.8)
H3K4me1_WTEE_track <- DataTrack(range=bw_H3K4me1_WTEE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="grey60",col.histogram="grey60",
                         name="H3K4me1 WTEE",
                         type="histogram", alpha=0.8)
H3K4me1_TGSE_track <- DataTrack(range=bw_H3K4me1_TGSE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="plum3",col.histogram="plum3",
                         name="H3K4me1 TGSE",
                         type="histogram", ,alpha=0.8)
H3K4me1_TGEE_track <- DataTrack(range=bw_H3K4me1_TGEE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="palegreen3",col.histogram="palegreen3",
                         name="H3K4me1 TGEE",
                         type="histogram", alpha=0.8)

#plotting - overlaid
H3K4me1_overlay_all <- OverlayTrack(trackList=list(H3K4me1_WTSE_track, H3K4me1_TGSE_track, H3K4me1_WTEE_track, H3K4me1_TGEE_track))
H3K4me1_overlay_SE <- OverlayTrack(trackList=list(H3K4me1_WTSE_track, H3K4me1_TGSE_track))
H3K4me1_overlay_EE <- OverlayTrack(trackList=list(H3K4me1_WTEE_track, H3K4me1_TGEE_track))

#plotting everything
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, H3K4me1_overlay_SE, H3K4me1_overlay_EE), from = 127205018-5000, to = 127488336+5000, sizes = c(0.05,0.1,0.05,0.1,0.1), col.axis="black", background.title="white")
```

# Plotting H3K27ac at Mgat5
For a representative example.

```{r Mgat5 plot, eval=F}
################## H3K27ac

#adding annotations
H3K27ac_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_conspeaks_annotated.csv")
H3K27ac_anno$annotation <- as.factor(H3K27ac_anno$annotation)
H3K27ac_anno$annotation <- reorder.factor(H3K27ac_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))

######## check stats for differential H3K27ac

#using dataframe from all conspeaks analysis and subsetting it
WTSE_TGSE_H3K27ac_all <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_WTSE-TGSE_anno.csv")
all(WTSE_TGSE_H3K27ac_all$coord %in% H3K27ac_anno$coord) #TRUE
WTSE_TGSE_H3K27ac_all[WTSE_TGSE_H3K27ac_all$SYMBOL=="Mgat5" & abs(WTSE_TGSE_H3K27ac_all$Fold)>=0.5 & WTSE_TGSE_H3K27ac_all$FDR<=0.05,c("seqnames","start","end","Fold", "FDR")]
#    seqnames     start       end  Fold     FDR
#39      chr1 127230995 127231822  3.16 0.00132
#590     chr1 127348788 127349231  2.46 0.02560
#675     chr1 127361161 127361886 -3.53 0.02970

all(WTSE_TGSE_H3K27ac_all$coord == H3K27ac_anno$coord) #TRUE
colnames(H3K27ac_anno)[5] <- "annotation_simple"
WTSE_TGSE_H3K27ac_all <- cbind(WTSE_TGSE_H3K27ac_all, H3K27ac_anno[,5:9])
WTSE_TGSE_H3K27ac_all[WTSE_TGSE_H3K27ac_all$SYMBOL=="Mgat5" & abs(WTSE_TGSE_H3K27ac_all$Fold)>=0.5 & WTSE_TGSE_H3K27ac_all$FDR<=0.05,c("coord","annotation_simple","chrom_anno_WTSE","chrom_anno_TGSE","chrom_anno_WTEE","chrom_anno_TGEE")]
#                        coord annotation_simple chrom_anno_WTSE chrom_anno_TGSE chrom_anno_WTEE chrom_anno_TGEE
#39  chr1: 127230995-127231822   Poised Enhancer         H3K4me1 H3K27ac_H3K4me1         H3K4me1         H3K4me1
#590 chr1: 127348788-127349231   Poised Enhancer            None H3K27ac_H3K4me1            None            None
#675 chr1: 127361161-127361886            Intron         H3K27ac            None            None            None

#first peak had only H3K4me1 in WTSE, and gained H3K27ac in TGSE, transitioning from poised --> active enhancer
#H3K27ac gain was unique to TGSE

#second peak gained both H3K4me1 and H3K27ac in TGSE

#third peak had H3K27ac in WTSE only, and lost it in TGSE

#what are the transitions over all genes with differential H3K27ac, H3K4me1, and DNAhm?
nrow(DBPs_H3K27ac <- WTSE_TGSE_H3K27ac_all[WTSE_TGSE_H3K27ac_all$SYMBOL%in%set16 & complete.cases(WTSE_TGSE_H3K27ac_all$SYMBOL) & abs(WTSE_TGSE_H3K27ac_all$Fold)>=0.5 & WTSE_TGSE_H3K27ac_all$FDR<=0.05,]) #22
DBPs_H3K27ac <- DBPs_H3K27ac %>% arrange(chrom_anno_WTSE)
DBPs_H3K27ac[,c("chrom_anno_WTSE","chrom_anno_TGSE")]
#   chrom_anno_WTSE chrom_anno_TGSE
#1          H3K27ac            None
#2          H3K27ac            None
#3          H3K27ac         H3K4me1
#4          H3K27ac            None
#5          H3K27ac            None
#6          H3K27ac            None
#7          H3K27ac            None
#8  H3K27ac_H3K4me1            None
#9  H3K27ac_H3K4me1         H3K4me1
#10 H3K27ac_H3K4me1 H3K27ac_H3K4me1
#11 H3K27ac_H3K4me1 H3K27ac_H3K4me1
#12 H3K27ac_H3K4me1 H3K27ac_H3K4me1
#13 H3K27ac_H3K4me1 H3K27ac_H3K4me1
#14 H3K27ac_H3K4me1 H3K27ac_H3K4me1
#15         H3K4me1 H3K27ac_H3K4me1
#16            None         H3K27ac
#17            None         H3K27ac
#18            None H3K27ac_H3K4me1
#19            None H3K27ac_H3K4me1
#20            None H3K27ac_H3K4me1
#21            None            None
#22            None         H3K27ac

DBPs_H3K27ac$H3K27ac_WTSE <- 0
DBPs_H3K27ac[grep("H3K27ac",DBPs_H3K27ac$chrom_anno_WTSE),"H3K27ac_WTSE"] <- 1
DBPs_H3K27ac$H3K27ac_TGSE <- 0
DBPs_H3K27ac[grep("H3K27ac",DBPs_H3K27ac$chrom_anno_TGSE),"H3K27ac_TGSE"] <- 1
DBPs_H3K27ac$H3K27ac_WTEE <- 0
DBPs_H3K27ac[grep("H3K27ac",DBPs_H3K27ac$chrom_anno_WTEE),"H3K27ac_WTEE"] <- 1
DBPs_H3K27ac$H3K27ac_TGEE <- 0
DBPs_H3K27ac[grep("H3K27ac",DBPs_H3K27ac$chrom_anno_TGEE),"H3K27ac_TGEE"] <- 1

DBPs_melt <- melt(DBPs_H3K27ac[,c("coord","Conc_WTSE","Conc_TGSE")], id.vars="coord")
DBPs_melt$variable <- gsub("Conc_", "", DBPs_melt$variable)
DBPs_melt$variable <- reorder.factor(DBPs_melt$variable, new.order=c("WTSE","TGSE"))

H3K27ac_conc <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/conc_H3K27ac_2.csv")
H3K27ac_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_WTEE-TGEE_anno.csv")
H3K27ac_TGEE <- H3K27ac_TGEE[H3K27ac_TGEE$coord %in% H3K27ac_conc$coord,]
H3K27ac_TGEE <- H3K27ac_TGEE[H3K27ac_TGEE$SYMBOL %in% set16,]
H3K27ac_TGEE <- H3K27ac_TGEE[complete.cases(H3K27ac_TGEE$SYMBOL),]

H3K27ac_conc <- H3K27ac_conc[match(H3K27ac_TGEE$coord, H3K27ac_conc$coord),]
all.equal(H3K27ac_conc$coord, H3K27ac_TGEE$coord) #TRUE
H3K27ac_conc$Gene <- H3K27ac_TGEE$SYMBOL

#adding annotations
H3K27ac_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K27ac/H3K27ac_conspeaks_annotated.csv")
H3K27ac_anno$annotation <- as.factor(H3K27ac_anno$annotation)
H3K27ac_anno$annotation <- reorder.factor(H3K27ac_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))
H3K27ac_anno <- H3K27ac_anno[match(H3K27ac_conc$coord, H3K27ac_anno$coord),]
all.equal(H3K27ac_anno$coord, H3K27ac_conc$coord) #TRUE
H3K27ac_conc$feature <- H3K27ac_anno$annotation
H3K27ac_conc <- H3K27ac_conc[,-1]

K27ac_melt <- melt(H3K27ac_conc, id.vars=c("coord","Gene","feature"))
K27ac_melt$group <- substr(K27ac_melt$variable, 1, 4)
K27ac_melt$group <- as.factor(K27ac_melt$group)
K27ac_melt$group <- reorder.factor(K27ac_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

ggplot(K27ac_melt[K27ac_melt$coord %in% DBPs_melt$coord,], aes(x=group, y=value, col=Gene)) + stat_summary(aes(group=K27ac_melt[K27ac_melt$coord %in% DBPs_melt$coord,"coord"]), fun.y=mean, geom="line") + geom_point() + theme_classic() + ylab("Concentration") + theme(axis.title.x=element_blank()) + ggtitle("H3K27ac differentially bound peaks (TGSE vs WTSE)")

H3K4me1_conc <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/conc_H3K4me1_2.csv")
H3K4me1_TGEE <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_WTEE-TGEE_anno.csv")
H3K4me1_TGEE <- H3K4me1_TGEE[H3K4me1_TGEE$coord %in% H3K4me1_conc$coord,]
H3K4me1_TGEE <- H3K4me1_TGEE[H3K4me1_TGEE$SYMBOL %in% set16,]
H3K4me1_TGEE <- H3K4me1_TGEE[complete.cases(H3K4me1_TGEE$SYMBOL),]

H3K4me1_conc <- H3K4me1_conc[match(H3K4me1_TGEE$coord, H3K4me1_conc$coord),]
all.equal(H3K4me1_conc$coord, H3K4me1_TGEE$coord) #TRUE
H3K4me1_conc$Gene <- H3K4me1_TGEE$SYMBOL

#adding annotations
H3K4me1_anno <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_all_conspeaks/H3K4me1/H3K4me1_conspeaks_annotated.csv")
H3K4me1_anno$annotation <- as.factor(H3K4me1_anno$annotation)
H3K4me1_anno$annotation <- reorder.factor(H3K4me1_anno$annotation, new.order=c("Poised Enhancer", "Active Enhancer", "Active Promoter", "Promoter", "5' UTR", "Exon", "Intron", "3' UTR", "Downstream", "Distal Intergenic"))
H3K4me1_anno <- H3K4me1_anno[match(H3K4me1_conc$coord, H3K4me1_anno$coord),]
all.equal(H3K4me1_anno$coord, H3K4me1_conc$coord) #TRUE
H3K4me1_conc$feature <- H3K4me1_anno$annotation
H3K4me1_conc <- H3K4me1_conc[,-1]

K4me1_melt <- melt(H3K4me1_conc, id.vars=c("coord","Gene","feature"))
K4me1_melt$group <- substr(K4me1_melt$variable, 1, 4)
K4me1_melt$group <- as.factor(K4me1_melt$group)
K4me1_melt$group <- reorder.factor(K4me1_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

length(DBPs_coord_gr <- GRanges(seqnames=DBPs_H3K27ac$seqnames, ranges=IRanges(start=DBPs_H3K27ac$start, end=DBPs_H3K27ac$end))) #22
length(K4me1_gr <- GRanges(seqnames=H3K4me1_anno$seqnames, ranges=IRanges(start=H3K4me1_anno$start, end=H3K4me1_anno$end))) #580
length(K4me1_gr <- subsetByOverlaps(K4me1_gr, DBPs_coord_gr)) #21
K4me1_coords <- paste(seqnames(K4me1_gr), ranges(K4me1_gr), sep=": ")

ggplot(K4me1_melt[K4me1_melt$coord %in% K4me1_coords,], aes(x=group, y=value, col=Gene)) + stat_summary(aes(group=K4me1_melt[K4me1_melt$coord %in% K4me1_coords,"coord"]), fun.y=mean, geom="line") + geom_point() + theme_classic() + ylab("H3K4me1 Concentration") + theme(axis.title.x=element_blank()) + ggtitle("H3K27ac differentially bound peaks (TGSE vs WTSE)")


#using dataframe with only sig peaks
WTSE_TGSE_H3K27ac <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_sig_only/H3K27ac/H3K27ac_WTSE-TGSE_cons-2_FDR0_05_FC0_5_annot.tsv")
WTSE_TGSE_H3K27ac$coord <- paste(WTSE_TGSE_H3K27ac$chr, paste(WTSE_TGSE_H3K27ac$start, WTSE_TGSE_H3K27ac$end, sep="-"), sep=": ")
all(WTSE_TGSE_H3K27ac$coord %in% H3K27ac_anno$coord) #FALSE
length(WTSE_TGSE_H3K27ac$coord[WTSE_TGSE_H3K27ac$coord %in% H3K27ac_anno$coord]) #0
WTSE_TGSE_H3K27ac[WTSE_TGSE_H3K27ac$SYMBOL=="Mgat5",c("chr","start","end","Fold", "FDR")]
#     chr     start       end  Fold     FDR
#39  chr1 127230996 127231822  3.16 0.00132
#581 chr1 127348789 127349231  2.46 0.02560
#665 chr1 127361162 127361886 -3.53 0.02970

#peak start is off by one base pair between the annotation and the tsv file

#does the chromatin state change between groups?
H3K27ac_anno[H3K27ac_anno$coord %in% c("chr1: 127230996-127231822", "chr1: 127348789-127349231", "chr1: 127361162-127361886"),]

#none for WTSE-WTEE, TGSE-TGEE

WTEE_TGEE_H3K27ac <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/diffbind_2_reps_sig_only/H3K27ac/H3K27ac_WTEE-TGEE_cons-2_FDR0_05_FC0_5_annot.tsv")
WTEE_TGEE_H3K27ac[WTEE_TGEE_H3K27ac$SYMBOL=="Mgat5" & complete.cases(WTEE_TGEE_H3K27ac$SYMBOL),c("chr","start","end","Fold", "FDR")]
#     chr     start       end  Fold   FDR
#929 chr1 127305118 127305930 -1.42 0.039


######## 

#UCSC annotation in mm10
#chr1:127205018-127488336

nrow(subbed_fdat_K4me1 <- H3K27ac_anno[H3K27ac_anno$seqnames=="chr1" & H3K27ac_anno$end>=127205018-5000 & H3K27ac_anno$start<=127488336+5000,]) #64 peaks
sort(summary(subbed_fdat_K4me1$annotation))
#         Promoter            5' UTR        Downstream Distal Intergenic   Active Promoter            3' UTR              Exon   Active Enhancer 
#                0                 0                 0                 0                 1                 1                 3                 6 
#           Intron   Poised Enhancer 
#                9                17 

# Chromosome Track
Mgat5_itrack <- IdeogramTrack(genome = "mm10", chromosome = "chr1")
# Basepair map track
Mgat5_gtrack <- GenomeAxisTrack(add53 = TRUE, add35 = TRUE, labelPos = "alternating")

#intron/exon track
#UCSC coords chr11:69098948-69109960
knownGenes <- UcscTrack(genome = "mm10", chromosome = "chr1", 
                        track = "knownGene", from = 127205018-5000, to = 127488336+5000,
                        trackType = "GeneRegionTrack", 
                        rstarts = "exonStarts", rends = "exonEnds", 
                        gene = "name", symbol = "name", 
                        transcript = "name", strand = "strand", 
                        fill = "grey", name = "UCSC Genes")

#### H3K27ac tracks
bw_H3K27ac_WTSE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K27ac/merged_b_WTSE.bw", as="GRanges")
bw_H3K27ac_WTEE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K27ac/merged_b_WTEE.bw", as="GRanges")
bw_H3K27ac_TGSE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K27ac/merged_b_TGSE.bw", as="GRanges")
bw_H3K27ac_TGEE <- import.bw("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/7-ChIPseq/Zinah_Wassouf_data/bigwig/H3K27ac/merged_b_TGEE.bw", as="GRanges")

#making trakcs
#optional argument: ylim=c(min,max)
H3K27ac_WTSE_track <- DataTrack(range=bw_H3K27ac_WTSE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="black",col.histogram="black",
                         name="H3K27ac WTSE",
                         type="histogram", alpha=0.8)
H3K27ac_WTEE_track <- DataTrack(range=bw_H3K27ac_WTEE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="grey60",col.histogram="grey60",
                         name="H3K27ac WTEE",
                         type="histogram", alpha=0.8)
H3K27ac_TGSE_track <- DataTrack(range=bw_H3K27ac_TGSE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="plum3",col.histogram="plum3",
                         name="H3K27ac TGSE",
                         type="histogram", ,alpha=0.8)
H3K27ac_TGEE_track <- DataTrack(range=bw_H3K27ac_TGEE,
                         chromosome="chr1",
                         from =127205018-5000,
                         to=127488336+5000,
                         fill="palegreen3",col.histogram="palegreen3",
                         name="H3K27ac TGEE",
                         type="histogram", alpha=0.8)

#plotting - overlaid
H3K27ac_overlay_all <- OverlayTrack(trackList=list(H3K27ac_WTSE_track, H3K27ac_TGSE_track, H3K27ac_WTEE_track, H3K27ac_TGEE_track))
H3K27ac_overlay_SE <- OverlayTrack(trackList=list(H3K27ac_WTSE_track, H3K27ac_TGSE_track))
H3K27ac_overlay_EE <- OverlayTrack(trackList=list(H3K27ac_WTEE_track, H3K27ac_TGEE_track))

#plotting everything
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127205018-5000, to = 127488336+5000, sizes = c(0.05,0.1,0.05,0.1,0.1), col.axis="black", background.title="white")

#zoomed in
#     chr     start       end  Fold     FDR
#39  chr1 127230996 127231822  3.16 0.00132
#581 chr1 127348789 127349231  2.46 0.02560
#665 chr1 127361162 127361886 -3.53 0.02970
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127230996, to = 127361886, sizes = c(0.05,0.1,0.05,0.1,0.1), col.axis="black", background.title="white")

##Plotting DNAhm, H3K4me1, and H3K27ac
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127205018-5000, to = 127488336+5000, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")

#differential DNAhm
#chr1.127378802
#chr1.127397681
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127378802-100, to = 127378802+100, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")

#differential H3K4me1
#      chr     start       end  Fold    FDR
#3337 chr1 127355656 127355959 -1.87 0.0331
#4268 chr1 127329211 127331204 -0.70 0.0449

plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127355656-100, to = 127355959+100, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127329211-100, to = 127331204+100, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")

#differential H3K27ac
#     chr     start       end  Fold     FDR
#39  chr1 127230996 127231822  3.16 0.00132
#581 chr1 127348789 127349231  2.46 0.02560
#665 chr1 127361162 127361886 -3.53 0.02970

plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127230996, to = 127231822, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127348789-100, to = 127349231+100, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")
plotTracks(list(Mgat5_itrack, Mgat5_gtrack, knownGenes, Mgat5_dTrack_hmC, H3K4me1_overlay_SE, H3K4me1_overlay_EE, H3K27ac_overlay_SE, H3K27ac_overlay_EE), from = 127361162, to = 127361886, sizes = c(0.05,0.1,0.05,0.1,0.1,0.1,0.1,0.1), col.axis="black", background.title="white")
```

---