DecipherPD Mouse Enriched Environment Hippocampus Differential Methylation
================================

##### Analyst: Samantha Schaffner
##### Date: Dec 5, 2022

Project summary:

This project investigates the effects of genotype and lifestyle factors on the DNA methylome of C57BL/6 mice. Specifically, wild type and human SNCA transgenic mice (with a BAC insertion containing full-length human SNCA and all its regulatory elements) were reared to the age of either 6 or 12 months before sacrificing. A subset of 6 month old mice from both genotypes were exposed to a chronic stress paradigm, while a subset of 12 month old mice from both genotypes were exposed to an enriched environment paradigm. From all groups, hippocampus and striatum tissues were collected.

DNA was extracted from mouse brain tissues and prepped for reduced representation bisulfite sequenicng (RRBS) using either a bisulfite conversion (mC + hmC) or oxidatitve bisulfite conversion (mC only) protocol. Samples were pooled in groups of 8 and sequenced on an Illumina HiSeq2500 (75bp, paired-end) at the BC Genome Sciences Centre. Alignment and methylation ratio calculations were performed using the BSMAP pipeline.

## Load in data and set working directories

```{r load data, eval=F}
.libPaths(c(.libPaths(),"/mnt/scratch/KoborLab/R_Libs/4.2"))
library(BiSeq)
library(GenomicRanges)
library(ggplot2)
library(reshape2)
library(dplyr)
library(gplots)
library(RColorBrewer)
library(gtools)
library(DescTools)
library(normtest)
library(ggpubr)
library(gridExtra)
library(bacon)
library(NCmisc)
library(limma)
library(car)
source("/mnt/scratch/KoborLab/shared_coding_resource/qqplot_NG.R")
```

## Differential methylation between groups: WTSE vs TGSE

```{r betaReg, eval=F}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpH/hippEE_oxBS_BSraw_CpH_10X_noout.RData")

#subset to autosomes
dim(oxBS_BSraw_CpH_noout) #3938054      13
unique(seqnames(oxBS_BSraw_CpH_noout)) #some chrY are there, measuring small amount of noise
# [1] chr1  chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chr2  chr3  chr4  chr5  chr6  chr7 
#[18] chr8  chr9  chrX  chrY 

dim(oxBS_BSraw_auto <- oxBS_BSraw_CpH_noout[-which(seqnames(oxBS_BSraw_CpH_noout) %in% c("chrX","chrY")),])
#3755512      13

#variability filter
BSrel_auto <- rawToRel(oxBS_BSraw_auto)
betas <- methLevel(BSrel_auto)

#Rachel's code for nonvariable 450k CpGs
sites <- paste(seqnames(oxBS_BSraw_auto), ranges(oxBS_BSraw_auto), sep=".")
betas_df <- as.data.frame(betas)
rownames(betas_df) <- sites
betas_df$chr <- paste(as.character(seqnames(oxBS_BSraw_auto)))
betas_df$pos <- paste(ranges(oxBS_BSraw_auto))

Variation<-function(x) {quantile(x, c(0.9), na.rm=T)[[1]]-quantile(x, c(0.1), na.rm=T)[[1]]}
ref_range<-sapply(1:nrow(betas), function(x) Variation(betas[x,]))
betas_variable <- betas_df[which(ref_range>=0.1),]
dim(betas_variable) #75,533 sites

##Subset your BSraw/BSrel objects to variable sites
variable_range <- GRanges(seqnames=betas_variable$chr, ranges=IRanges(start=as.numeric(betas_variable$pos), end=as.numeric(betas_variable$pos)))

dim(BSraw_auto_var <- subsetByOverlaps(oxBS_BSraw_auto, variable_range)) #75533    13

#matching meta data to colnames of BSraw object
meta <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/mapq_filtered_qc_stats.csv")
meta <- meta[meta$Conversion=="oxBS" & meta$Age_months==12,]
meta <- meta[-which(meta$Pool=="PX1175"),]
#fill in "SampleID" column: numbered 1-4 by adapter (alphabetical)
meta$SampleID <- paste(meta$Genotype, meta$Environment, sep="")
meta$SampleID <- gsub("STD", "SE", meta$SampleID)
meta <- meta %>% arrange(Adapter) %>% arrange(desc(Environment)) %>% arrange(desc(Genotype))
meta$SampleID <- paste(meta$SampleID, rep(seq(1:4),4), sep="")
meta <- meta[match(colnames(BSraw_auto_var), meta$SampleID),]
all(meta$SampleID==colnames(BSraw_auto_var)) #TRUE

#replace colData slot in BSraw object and save
colData(BSraw_auto_var) <- DataFrame(meta)
colnames(BSraw_auto_var) <- meta$SampleID
save(BSraw_auto_var, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpH/oxBS_BSraw_CpH_10X_noout_auto_var.RData")

betas <- methLevel(rawToRel(BSraw_auto_var))

Beta_raw_sample_melted<- melt(betas)
Beta_raw_Plot<-Beta_raw_sample_melted[which(Beta_raw_sample_melted$value >= 0),]
ggplot(Beta_raw_Plot, aes(value, group=Var2, color=Var2))+ geom_density(size=1)+theme_bw() + ggtitle("CpH methylation (coverage- and variability-filtered)") + theme(legend.position = "none")

sites <- paste(seqnames(BSraw_auto_var), ranges(BSraw_auto_var), sep=".")
rownames(betas) <- sites

colData <- colData(BSraw_auto_var)
all(colData$SampleID==colnames(betas)) #TRUE
str(colData) #check G/E reference levels

vif(lm(betas[1,] ~ colData$Genotype*colData$Environment))
#                    colData$Genotype                  colData$Environment 
#                            2.153846                             2.153846 
#colData$Genotype:colData$Environment 
#                            3.461538 

vif(lm(betas[1,] ~ colData$Genotype+colData$Environment))
#   colData$Genotype colData$Environment 
#           1.005128            1.005128 

vif(lm(betas[1,] ~ colData$Genotype*colData$Environment),type="predictor")
#               GVIF Df GVIF^(1/(2*Df))      Interacts With               Other Predictors
#colData     3.47929  0             Inf colData$Environment colData, Genotype, Environment
#Genotype    3.47929  0             Inf colData$Environment colData, Genotype, Environment
#Environment 3.47929  0             Inf    colData$Genotype colData, Genotype, Environment

vif(lm(betas[1,] ~ colData$Genotype+colData$Environment),type="predictor")
#                GVIF Df GVIF^(1/(2*Df))      Interacts With               Other Predictors
#colData     1.005128  0             Inf colData$Environment colData, Genotype, Environment
#Genotype    1.005128  0             Inf                --   colData, Genotype, Environment
#Environment 1.005128  0             Inf                --   colData, Genotype, Environment

str(colData$Genotype)
str(colData$Environment)
colData$Genotype <- as.factor(colData$Genotype)
colData$Genotype <- reorder.factor(colData$Genotype, new.order=c("WT","TG"))
colData$Environment <- as.factor(colData$Environment)
colData$Environment <- reorder.factor(colData$Environment, new.order=c("STD","EE"))

#LM on variable cytosines
lm_results_G <- lapply(1:nrow(betas), function(x) {
  return(summary(lm(betas[x,] ~ colData$Genotype*colData$Environment))$coefficients[2,])
})
results_G <- as.data.frame(do.call(rbind, lm_results_G))
results_G$z <- zscoreT(results_G$`t value`, df=12)
results_G$site <- rownames(betas)
results_G_complete <- results_G[complete.cases(results_G$z),]
bcz <- bacon(results_G_complete$z)
#Bacon-object containing 1 set(s) of 75533 test-statistics.
#...estimated bias: -0.18.
#...estimated inflation: 0.97.
#
#Empirical null estimates are based on 5000 iterations with a burnin-period of 2000.

hist(results_G$`Pr(>|t|)`)
qqunif.plot(results_G$`Pr(>|t|)`[complete.cases(results_G$`Pr(>|t|)`)], title="lm ~Genotype*Environment")

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_GxE_CpH_auto.png", width=305, height=335)
ggplot(data=results_G_complete, aes(x=Estimate, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()

#calculate delta betas manually
results_G$site <- rownames(betas)
colnames(betas)
results_G$DB <- rowMeans(betas[,8:10]) - rowMeans(betas[,1:4])

write.csv(results_G, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/lm_WTSE_TGSE_CpH.csv", row.names=F)

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_GxE_CpH_auto_manual_DB.png", width=305, height=335)
ggplot(data=results_G, aes(x=DB, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()
```
![P-value histogram](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/pval_hist_CpH_auto.png)

![Q-Q plot](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/qqplot_CpH_auto.png)

![Volcano plot](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_GxE_CpH_auto.png)

## Thresholded volcano plot 

Using GxE results. "Estimate" and manual delta beta should be the same.

```{r volcano 1, eval=F}
results_G_complete <- read.csv("~/KoborLab/Projects/DecipherPD_mouse/EE_asyn/sschaffner/3-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/auto/CpH/lm_WTSE_TGSE_CpH.csv")
cor(results_G_complete$Estimate, results_G_complete$DB) # 1
results_G_complete$adjP <- p.adjust(results_G_complete$Pr...t.., method="BH")

#Volcano plot
results_G_complete$threshold <- (results_G_complete$adjP<=0.05 & abs(results_G_complete$Estimate)>=0.1) 

#annotating which hits have increased/decreased DNAm or no change
results_G_complete$DNAm_change <- results_G_complete$threshold
results_G_complete[results_G_complete$threshold==TRUE & results_G_complete$Estimate>0,]$DNAm_change <- "Increase"
results_G_complete[results_G_complete$threshold==TRUE & results_G_complete$Estimate<0,]$DNAm_change <- "Decrease"
results_G_complete[results_G_complete$threshold==FALSE,]$DNAm_change <- "NS"
summary(as.factor(results_G_complete$DNAm_change))
#Decrease Increase       NS 
#       2       12    75519 

#testing skewness of delta betas
skewness.norm.test(results_G_complete$Estimate, nrepl=1000)
#data:  results_G_complete$Estimate
#T = -0.0011581, p-value = 0.887

#get the nominal p threshold which corresponds to your FDR cutoff
max(results_G_complete[results_G_complete$threshold==TRUE,]$`Pr...t..`) #8.287476e-06

#Code for plotting. If you want to plot adjusted p on the y axis, ensure to change the variable and axis label. Also check that the numbers for your pcutoff threshold and delta beta threshold are correct (for plotting cutoff lines).
library(ggrepel)

png("~/KoborLab/Projects/DecipherPD_mouse/EE_asyn/sschaffner/3-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/auto/CpH/volcano_WTSE_TGSE_CpH_thresholded_nomP_0.05.png", width=305, height=335)
ggplot(data=results_G_complete, aes(x=Estimate, y=-log10(`Pr...t..`), colour=DNAm_change)) +
  geom_point(size=3) +
  labs(legend.position = "none") +
  xlab("Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray60","gray60", "gray87")) + geom_hline(yintercept=-log10(8.287476e-06)) + geom_vline(xintercept=-0.1) + geom_vline(xintercept=0.1) + theme(legend.position = "none", text=element_text(size=20)) + xlim(c(-0.8,0.8)) + ylim(c(0,15)) + geom_label_repel(data = results_G_complete[results_G_complete$threshold==TRUE,], aes(label = gene), size = 8, force = 10, max.overlaps = 30, min.segment.length=0)
dev.off()

#annotate genes
anno <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpH/auto/10X_filter/CpH_anno_mC_ChIP.txt")
anno$site <- gsub(": ", ".", anno$site)
all(anno$site==results_G_complete$site) #TRUE
results_G_complete <- cbind(results_G_complete, anno[,2:5])
write.csv(results_G_complete, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/auto/CpH/lm_WTSE_TGSE_CpH.csv", row.names=F)

hits <- results_G_complete[results_G_complete$threshold==TRUE,c(6,11,13,14,7,8)]
write.csv(hits, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/auto/CpH/lm_WTSE_TGSE_mC_CpH_hits.csv", row.names=F)
```
![Volcano plot with thresholds](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/auto/CpH/volcano_WTSE_TGSE_CpH_thresholded_nomP.png)

### Box plots
```{r box plots, eval=F}
nrow(betas_sub <- as.data.frame(betas[rownames(betas) %in% hits[hits$adjP<=0.05,"site"],])) #14

betas_sub$CpG <- rownames(betas_sub)
betas_melt <- melt(betas_sub, id.vars="CpG")
betas_melt$group <- substr(betas_melt$variable, start=1, stop=4)
betas_melt$group <- reorder.factor(betas_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

ggplot(betas_melt, aes(x=group,y=value,col=group)) + scale_colour_manual(values=c("black","plum3","gray60","palegreen3")) + theme_classic() + facet_wrap(~CpG) + geom_point() + geom_boxplot() + ylab("Beta value") + theme(axis.title.x=element_blank()) + ylim(c(0,1))

#check sig in each comparison
hits[hits$adjP<=0.05,c("site","adjP")]
#all adjP are between 0.017-0.045

EE_results <- read.csv("~/KoborLab/Projects/DecipherPD_mouse/EE_asyn/sschaffner/3-methdiff_DNAm_DNAhm/methylation/WTEE_TGEE/linear_model/auto/CpH/lm_WTEE_TGEE_CpH.csv")
EE_results$adjP <- p.adjust(EE_results$Pr...t.., method="BH")
EE_results[EE_results$site %in% hits[hits$adjP<=0.05,"site"] & EE_results$adjP<=0.05,c("site","adjP")]
#                site        adjP
#4422  chr1.177421430 0.027010572
#6794  chr10.65938345 0.023709136
#20415 chr13.99788771 0.028353712
#22242 chr14.53929892 0.021828734
#34160 chr18.51073430 0.001518654
#58730  chr6.15971331 0.007575420

WT_results <- read.csv("~/KoborLab/Projects/DecipherPD_mouse/EE_asyn/sschaffner/3-methdiff_DNAm_DNAhm/methylation/WTSE_WTEE/linear_model/CpH/lm_WTSE_WTEE.csv")
WT_results[WT_results$site %in% hits[hits$adjP<=0.05,"site"] & WT_results$adjP<=0.05,c("site","adjP")] #none

TG_results <- read.csv("~/KoborLab/Projects/DecipherPD_mouse/EE_asyn/sschaffner/3-methdiff_DNAm_DNAhm/methylation/TGSE_TGEE/linear_model/CpH/lm_TGSE_TGEE.csv")
TG_results[TG_results$site %in% hits[hits$adjP<=0.05,"site"] & TG_results$adjP<=0.05,c("site","adjP")] #none
#                site       adjP
#11912 chr11.84870406 0.01596879
#51770 chr4.140795682 0.04302577
#58747  chr6.16966299 0.02259320
```


