DecipherPD Mouse Enriched Environment Hippocampus Differential Methylation
================================

##### Analyst: Samantha Schaffner
##### Date: Nov 7, 2022

Project summary:

This project investigates the effects of genotype and lifestyle factors on the DNA methylome of C57BL/6 mice. Specifically, wild type and human SNCA transgenic mice (with a BAC insertion containing full-length human SNCA and all its regulatory elements) were reared to the age of either 6 or 12 months before sacrificing. A subset of 6 month old mice from both genotypes were exposed to a chronic stress paradigm, while a subset of 12 month old mice from both genotypes were exposed to an enriched environment paradigm. From all groups, hippocampus and striatum tissues were collected.

DNA was extracted from mouse brain tissues and prepped for reduced representation bisulfite sequenicng (RRBS) using either a bisulfite conversion (mC + hmC) or oxidatitve bisulfite conversion (mC only) protocol. Samples were pooled in groups of 8 and sequenced on an Illumina HiSeq2500 (75bp, paired-end) at the BC Genome Sciences Centre. Alignment and methylation ratio calculations were performed using the BSMAP pipeline.

## Load in data and set working directories

```{r load data, eval=F}
.libPaths(c(.libPaths(),"/mnt/scratch/KoborLab/R_Libs/4.2"))
library(BiSeq)
library(GenomicRanges)
library(ggplot2)
library(reshape2)
library(dplyr)
library(gplots)
library(RColorBrewer)
library(gtools)
library(DescTools)
library(normtest)
library(ggpubr)
library(gridExtra)
library(bacon)
library(NCmisc)
library(limma)
library(car)
source("/mnt/scratch/KoborLab/shared_coding_resource/qqplot_NG.R")
```

## Differential methylation between groups: WTSE vs TGSE

```{r betaReg, eval=F}
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpG/hippEE_oxBS_BSraw_CpG_10X_noout.RData")

betas <- methLevel(rawToRel(oxBS_BSraw_CpG_noout))

Beta_raw_sample_melted<- melt(betas)
Beta_raw_Plot<-Beta_raw_sample_melted[which(Beta_raw_sample_melted$value >= 0),]
ggplot(Beta_raw_Plot, aes(value, group=Var2, color=Var2))+ geom_density(size=1)+theme_bw() + ggtitle("CpG methylation (coverage-filtered)") + theme(legend.position = "none")

#subset to autosomes
dim(oxBS_BSraw_CpG_noout) #914215     13
unique(seqnames(oxBS_BSraw_CpG_noout)) #some chrY are there, measuring small amount of noise
# [1] chr1  chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chr2  chr3  chr4  chr5  chr6  chr7 
#[18] chr8  chr9  chrX  chrY 

dim(oxBS_BSraw_auto <- oxBS_BSraw_CpG_noout[-which(seqnames(oxBS_BSraw_CpG_noout) %in% c("chrX","chrY")),])
#882506     13

#variability filter
BSrel_auto <- rawToRel(oxBS_BSraw_auto)
betas <- methLevel(BSrel_auto)

Beta_raw_sample_melted<- melt(betas)
Beta_raw_Plot<-Beta_raw_sample_melted[which(Beta_raw_sample_melted$value >= 0),]
ggplot(Beta_raw_Plot, aes(value, group=Var2, color=Var2))+ geom_density(size=1)+theme_bw() + ggtitle("CpG methylation (coverage-filtered)") + theme(legend.position = "none")

#Rachel's code for nonvariable 450k CpGs
sites <- paste(seqnames(oxBS_BSraw_auto), ranges(oxBS_BSraw_auto), sep=".")
betas_df <- as.data.frame(betas)
rownames(betas_df) <- sites
betas_df$chr <- paste(as.character(seqnames(oxBS_BSraw_auto)))
betas_df$pos <- paste(ranges(oxBS_BSraw_auto))

Variation<-function(x) {quantile(x, c(0.9), na.rm=T)[[1]]-quantile(x, c(0.1), na.rm=T)[[1]]}
ref_range<-sapply(1:nrow(betas), function(x) Variation(betas[x,]))
betas_variable <- betas_df[which(ref_range>=0.1),]
dim(betas_variable) #329,507 sites

##Subset your BSraw/BSrel objects to variable sites
variable_range <- GRanges(seqnames=betas_variable$chr, ranges=IRanges(start=as.numeric(betas_variable$pos), end=as.numeric(betas_variable$pos)))

dim(BSraw_auto_var <- subsetByOverlaps(oxBS_BSraw_auto, variable_range)) #329507     13

#matching meta data to colnames of BSraw object
meta <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/mapq_filtered_qc_stats.csv")
meta <- meta[meta$Conversion=="oxBS" & meta$Age_months==12,]
meta <- meta[-which(meta$Pool=="PX1175"),]
#fill in "SampleID" column: numbered 1-4 by adapter (alphabetical)
meta$SampleID <- paste(meta$Genotype, meta$Environment, sep="")
meta$SampleID <- gsub("STD", "SE", meta$SampleID)
meta <- meta %>% arrange(Adapter) %>% arrange(desc(Environment)) %>% arrange(desc(Genotype))
meta$SampleID <- paste(meta$SampleID, rep(seq(1:4),4), sep="")
meta <- meta[match(colnames(BSraw_auto_var), meta$SampleID),]
all(meta$SampleID==colnames(BSraw_auto_var)) #TRUE

#replace colData slot in BSraw object and save
colData(BSraw_auto_var) <- DataFrame(meta)
colnames(BSraw_auto_var) <- meta$SampleID
save(BSraw_auto_var, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpG/oxBS_BSraw_CpG_10X_noout_auto_var.RData")

betas <- methLevel(rawToRel(BSraw_auto_var))

Beta_raw_sample_melted<- melt(betas)
Beta_raw_Plot<-Beta_raw_sample_melted[which(Beta_raw_sample_melted$value >= 0),]
ggplot(Beta_raw_Plot, aes(value, group=Var2, color=Var2))+ geom_density(size=1)+theme_bw() + ggtitle("CpG methylation (coverage- and variability-filtered)") + theme(legend.position = "none")

sites <- paste(seqnames(BSraw_auto_var), ranges(BSraw_auto_var), sep=".")
rownames(betas) <- sites

colData <- colData(BSraw_auto_var)
all(colData$SampleID==colnames(betas)) #TRUE
str(colData) #check G/E reference levels

vif(lm(betas[1,] ~ colData$Genotype*colData$Environment))
#                    colData$Genotype                  colData$Environment 
#                            2.153846                             2.153846 
#colData$Genotype:colData$Environment 
#                            3.461538 

vif(lm(betas[1,] ~ meta$Genotype*meta$Environment+meta$Percent_aligned)
#                 meta$Genotype               meta$Environment           meta$Percent_aligned meta$Genotype:meta$Environment 
#                      2.233159                       2.432681                       1.467469                       3.468177 

AIC(lm(betas[1,] ~ meta$Genotype*meta$Environment)) #[1] -13.15
AIC(lm(betas[1,] ~ meta$Genotype*meta$Environment + meta$Percent_aligned)) # -12.99533

vif(lm(betas[1,] ~ colData$Genotype+colData$Environment))
#   colData$Genotype colData$Environment 
#           1.005128            1.005128 

vif(lm(betas[1,] ~ colData$Genotype*colData$Environment),type="predictor")
#               GVIF Df GVIF^(1/(2*Df))      Interacts With               Other Predictors
#colData     3.47929  0             Inf colData$Environment colData, Genotype, Environment
#Genotype    3.47929  0             Inf colData$Environment colData, Genotype, Environment
#Environment 3.47929  0             Inf    colData$Genotype colData, Genotype, Environment

vif(lm(betas[1,] ~ colData$Genotype+colData$Environment),type="predictor")
#                GVIF Df GVIF^(1/(2*Df))      Interacts With               Other Predictors
#colData     1.005128  0             Inf colData$Environment colData, Genotype, Environment
#Genotype    1.005128  0             Inf                --   colData, Genotype, Environment
#Environment 1.005128  0             Inf                --   colData, Genotype, Environment

vif(lm(betas[1,] ~ meta$Genotype*meta$Environment+meta$Percent_aligned),type="predictor")
#                    GVIF Df GVIF^(1/(2*Df))                         Interacts With                             Other Predictors
#meta            5.105751  0             Inf meta$Environment, meta$Percent_aligned meta, Genotype, Environment, Percent_aligned
#Genotype        5.105751  0             Inf                       meta$Environment meta, Genotype, Environment, Percent_aligned
#Environment     5.105751  0             Inf                          meta$Genotype meta, Genotype, Environment, Percent_aligned
#Percent_aligned 5.105751  0             Inf                                   --   meta, Genotype, Environment, Percent_aligned
                      
str(colData$Genotype)
str(colData$Environment)
colData$Genotype <- as.factor(colData$Genotype)
colData$Genotype <- reorder.factor(colData$Genotype, new.order=c("WT","TG"))
colData$Environment <- as.factor(colData$Environment)
colData$Environment <- reorder.factor(colData$Environment, new.order=c("STD","EE"))

#LM on variable cytosines
lm_results_G <- lapply(1:nrow(betas), function(x) {
  return(summary(lm(betas[x,] ~ colData$Genotype*colData$Environment))$coefficients[2,])
})
results_G <- as.data.frame(do.call(rbind, lm_results_G))
results_G$z <- zscoreT(results_G$`t value`, df=12)
results_G$site <- rownames(betas)
results_G_complete <- results_G[complete.cases(results_G$z),]
bcz <- bacon(results_G_complete$z)
#Bacon-object containing 1 set(s) of 329507 test-statistics.
#...estimated bias: -0.06.
#...estimated inflation: 1.
#
#Empirical null estimates are based on 5000 iterations with a burnin-period of 2000.

hist(results_G$`Pr(>|t|)`)
qqunif.plot(results_G$`Pr(>|t|)`[complete.cases(results_G$`Pr(>|t|)`)], title="lm ~Genotype*Environment")

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_GxE_CpG_auto.png", width=305, height=335)
ggplot(data=results_G_complete, aes(x=Estimate, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()

#calculate delta betas manually
results_G$site <- rownames(betas)
colnames(betas)
results_G$DB <- rowMeans(betas[,8:10]) - rowMeans(betas[,1:4])

write.csv(results_G, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/lm_WTSE_TGSE.csv", row.names=F)

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_GxE_CpG_auto_manual_DB.png", width=305, height=335)
ggplot(data=results_G, aes(x=DB, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()

#LM on variable cytosines, correcting for percent aligned
lm_results_G <- lapply(1:nrow(betas), function(x) {
  return(summary(lm(betas[x,] ~ colData$Genotype*colData$Environment+colData$Percent_aligned))$coefficients[2,])
})
results_G <- as.data.frame(do.call(rbind, lm_results_G))
results_G$z <- zscoreT(results_G$`t value`, df=12)
results_G$site <- rownames(betas)
results_G_complete <- results_G[complete.cases(results_G$z),]
bcz <- bacon(results_G_complete$z)
#...estimated bias: -0.049.
#...estimated inflation: 1.

hist(results_G$`Pr(>|t|)`)
qqunif.plot(results_G$`Pr(>|t|)`[complete.cases(results_G$`Pr(>|t|)`)], title="lm ~Genotype*Environment")

#RLM on variable cytosines
library(MASS)
library(sfsmisc)
rlm_results_G <- lapply(1:nrow(betas), function(x) {
  RLM <- rlm(betas[x,] ~ colData$Genotype*colData$Environment, psi=psi.huber, maxit=1000)
  return(f.robftest(RLM, var = names(RLM$coefficients)[2])$p.value)
})
#Warning in rlm.default(x, y, weights, method = method, wt.method = wt.method,  :
#  'rlm' failed to converge in 1000 steps
#Error in solve.default(R, rinv) : 
#  system is computationally singular: reciprocal condition number = 1.03688e-16

##### G + E
#LM on variable cytosines
lm_results_G_E <- lapply(1:nrow(betas), function(x) {
  return(summary(lm(betas[x,] ~ colData$Genotype+colData$Environment))$coefficients[2,])
})
results_G_E <- as.data.frame(do.call(rbind, lm_results_G_E))
results_G_E$z <- zscoreT(results_G_E$`t value`, df=12)
results_G_E$site <- rownames(betas)
results_G_E_complete <- results_G_E[complete.cases(results_G_E$z),]
bcz <- bacon(results_G_E_complete$z)
#Bacon-object containing 1 set(s) of 329507 test-statistics.
#...estimated bias: -0.045.
#...estimated inflation: 1.
#
#Empirical null estimates are based on 5000 iterations with a burnin-period of 2000.

hist(results_G_E$`Pr(>|t|)`)
qqunif.plot(results_G_E$`Pr(>|t|)`[complete.cases(results_G_E$`Pr(>|t|)`)], title="lm ~Genotype+Environment")

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_G+E_CpG_auto.png", width=305, height=335)
ggplot(data=results_G_E_complete, aes(x=Estimate, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()

#calculate delta betas manually
results_G_E$site <- rownames(betas)
colnames(betas)
results_G_E$DB <- rowMeans(betas[,8:13]) - rowMeans(betas[,1:7])

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_G+E_CpG_auto_manual_DB_allTG.png", width=305, height=335)
ggplot(data=results_G_E, aes(x=DB, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()

##### G
#LM on variable cytosines
lm_results_G <- lapply(1:nrow(betas), function(x) {
  return(summary(lm(betas[x,] ~ colData$Genotype))$coefficients[2,])
})
results_G <- as.data.frame(do.call(rbind, lm_results_G))
results_G$z <- zscoreT(results_G$`t value`, df=12)
results_G$site <- rownames(betas)
results_G_complete <- results_G[complete.cases(results_G$z),]
bcz <- bacon(results_G_complete$z)
#Bacon-object containing 1 set(s) of 329507 test-statistics.
#...estimated bias: -0.05.
#...estimated inflation: 1.
#
#Empirical null estimates are based on 5000 iterations with a burnin-period of 2000.

hist(results_G$`Pr(>|t|)`)
qqunif.plot(results_G$`Pr(>|t|)`[complete.cases(results_G$`Pr(>|t|)`)], title="lm ~Genotype")

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_G_CpG_auto.png", width=305, height=335)
ggplot(data=results_G_complete, aes(x=Estimate, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()

#calculate delta betas manually
results_G$site <- rownames(betas)
colnames(betas)
results_G$DB <- rowMeans(betas[,8:13]) - rowMeans(betas[,1:7])

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_G_CpG_auto_manual_DB_allTG.png", width=305, height=335)
ggplot(data=results_G, aes(x=DB, y=-log10(`Pr(>|t|)`), colour="grey")) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c( "gray87")) + theme(legend.position = "none")
dev.off()
```
![P-value histogram](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/pval_hist_WTSE_TGSE_lm.png)

![Q-Q plot](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/qqplot_WTSE_TGSE_lm.png)
![Volcano plot](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_WTSE_TGSE_lm.png)

## Thresholded volcano plot 

Using GxE results. "Estimate" and manual delta beta should be the same.

```{r volcano 1, eval=F}
results_G <- read.csv("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/lm_WTSE_TGSE.csv")
results_G_complete <- results_G[complete.cases(results_G$`Pr(>|t|)`),]
cor(results_G_complete$Estimate, results_G_complete$DB) # 1
results_G_complete$adjP <- p.adjust(results_G_complete$`Pr(>|t|)`, method="BH")

#Volcano plot
results_G_complete$threshold <- (results_G_complete$adjP<=0.15 & abs(results_G_complete$Estimate)>=0.1) #none

#annotating which hits have increased/decreased DNAm or no change
results_G_complete$DNAm_change <- results_G_complete$threshold
results_G_complete[results_G_complete$threshold==TRUE & results_G_complete$Estimate>0,]$DNAm_change <- "Increase"
results_G_complete[results_G_complete$threshold==TRUE & results_G_complete$Estimate<0,]$DNAm_change <- "Decrease"
results_G_complete[results_G_complete$threshold==FALSE,]$DNAm_change <- "NS"
summary(as.factor(results_G_complete$DNAm_change))
#Decrease Increase       NS 
#       3        2   329502 

#testing skewness of delta betas
skewness.norm.test(results_G_complete$Estimate, nrepl=1000)
#data:  results_G_complete$Estimate
#T = 0.02037, p-value < 2.2e-16

#get the nominal p threshold which corresponds to your FDR cutoff
max(results_G_complete[results_G_complete$threshold==TRUE,]$`Pr...t..`) #1.584301e-06

#Code for plotting. If you want to plot adjusted p on the y axis, ensure to change the variable and axis label. Also check that the numbers for your pcutoff threshold and delta beta threshold are correct (for plotting cutoff lines).
library(ggrepel)
png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_WTSE_TGSE_CpG_thresholded.png", width=305, height=335)
ggplot(data=results_G_complete, aes(x=Estimate, y=-log10(adjP), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta Beta") + ylab("-log10 adj P-Value") + theme_classic() + scale_color_manual(values=c("gray60", "gray87")) + geom_hline(yintercept=-log10(0.05)) + geom_vline(xintercept=-0.1) + geom_vline(xintercept=0.1) + theme(legend.position = "none") 
dev.off()

#annotate genes
anno <- read.delim("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpG/cpg_anno_mC_ChIP.txt")
anno$site <- gsub(": ", ".", anno$site)
all(anno$site==results_G_complete$site) #TRUE
results_G_complete <- cbind(results_G_complete, anno[,2:5])
write.csv(results_G_complete, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/CpG/lm_WTSE_TGSE.csv", row.names=F)

summary(results_G_complete$Estimate)
max(-log10(results_G_complete$Pr...t..))

png("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/CpG/volcano_WTSE_TGSE_CpG_thresholded_nomP.png", width=305, height=335)
ggplot(data=results_G_complete, aes(x=Estimate, y=-log10(`Pr...t..`), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Delta Beta") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray60", "gray60","gray87")) + geom_hline(yintercept=-log10(1.584301e-06)) + geom_vline(xintercept=-0.1) + geom_vline(xintercept=0.1) + theme(legend.position = "none")   + geom_label_repel(data = results_G_complete[results_G_complete$threshold==TRUE,], aes(label = gene), size = 4, force = 10, max.overlaps = 30, min.segment.length=0) + xlim(c(-0.75,0.75)) + ylim(c(0,12))
dev.off()

write.csv(results_G_complete, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/lm_WTSE_TGSE.csv", row.names=F)

hits <- results_G_complete[results_G_complete$threshold==TRUE,c(6,11,13,14,7,8)]
write.csv(hits, file="/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/CpG/lm_WTSE_TGSE_mC_hits.csv", row.names=F)
```
![Volcano plot with thresholds](/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/methylation/WTSE_TGSE/linear_model/volcano_WTSE_TGSE_CpG_thresholded_nomP.png)
## Boxplots
```{r}
CpGs <- hits$site
load("/mnt/scratch/KoborLab/DecipherPD_mouse/ee_asyn/hipp/sschaffner/082022/5-methdiff_DNAm_DNAhm/preprocessing/oxbisulfite/CpG/all_EE_mC_betas_CpG_10X.RData")
colnames(all_EE)
all_EE <- all_EE[,-c(8,12,13)]
all_EE_sub <- all_EE[rownames(all_EE) %in% CpGs,]
all_EE_sub$CpG <- rownames(all_EE_sub)
betas_melt <- melt(all_EE_sub, id.vars="CpG")
betas_melt$group <- substr(betas_melt$variable, start=1, stop=4)
betas_melt$group <- reorder.factor(betas_melt$group, new.order=c("WTSE","TGSE","WTEE","TGEE"))

ggplot(betas_melt, aes(x=group,y=value,col=group)) + scale_colour_manual(values=c("black","plum3","gray60","palegreen3")) + theme_classic() + facet_wrap(~CpG) + geom_point() + geom_boxplot() + ylab("Beta value") + theme(axis.title.x=element_blank())
```
